From fc248814fe827edffd971ae8b14969d6fa05a0e0 Mon Sep 17 00:00:00 2001
From: Juanjo <juanjo.ng.83@gmail.com>
Date: Mon, 14 Oct 2013 17:49:32 +0000
Subject: [PATCH 0063/1053] Reset queued command when changing CTRL state.

---
 src/toolbar_gui.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/toolbar_gui.cpp b/src/toolbar_gui.cpp
index 20f66d98a4..0226d61dce 100644
--- a/src/toolbar_gui.cpp
+++ b/src/toolbar_gui.cpp
@@ -2259,6 +2259,7 @@ struct TabletToolbar : Window {
 				this->ToggleWidgetLoweredState(WID_TT_CTRL);
 				HandleCtrlChanged();
 				this->SetWidgetDirty(WID_TT_CTRL);
+				EraseQueuedTouchCommand();
 				break;
 			case WID_TT_MOVE:
 				_move_pressed = !_move_pressed;
-- 
2.55.0

