From 571b244680cd61b607928a6cffbb16cc20a39ca5 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 20 Mar 2014 17:18:47 +0200
Subject: [PATCH 0088/1053] Updated todo

---
 todo.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/todo.txt b/todo.txt
index 9e4a566ec1..c0bb672060 100644
--- a/todo.txt
+++ b/todo.txt
@@ -4,3 +4,4 @@
 - Fix Online Content dialog
 - Fix AI dialog
 - Fix Multiplayer dialog
+- Fix missing sprites error
-- 
2.55.0

