From 6a81df3ecb06377f6e7e8ff6ac6020f079b4ae14 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 20 Mar 2014 23:42:14 +0200
Subject: [PATCH 0092/1053] Updated todo

---
 todo.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/todo.txt b/todo.txt
index 885d13f3cb..946c6d5998 100644
--- a/todo.txt
+++ b/todo.txt
@@ -3,3 +3,4 @@
 - Fix Online Content dialog
 - Fix AI dialog
 - Fix Multiplayer dialog
+- Scrollbar slider may become too narrow
-- 
2.55.0

