From 6bdb30fea581ac5ddcf5d20018be7baa600799c6 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 22 Mar 2014 01:44:41 +0200
Subject: [PATCH 0096/1053] Fix to file selector GUI

---
 src/fios_gui.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fios_gui.cpp b/src/fios_gui.cpp
index ced9e722d8..795a177575 100644
--- a/src/fios_gui.cpp
+++ b/src/fios_gui.cpp
@@ -355,14 +355,14 @@ public:
 			case WID_SL_DRIVES_DIRECTORIES_LIST: {
 				GfxFillRect(r.left + 1, r.top + 1, r.right, r.bottom, PC_BLACK);
 
-				uint y = r.top + WD_FRAMERECT_TOP + this->resize.step_height / 2;
+				uint y = r.top + WD_FRAMERECT_TOP;
 				for (uint pos = this->vscroll->GetPosition(); pos < _fios_items.Length(); pos++) {
 					const FiosItem *item = _fios_items.Get(pos);
 
 					if (item == this->selected) {
 						GfxFillRect(r.left + 1, y, r.right, y + this->resize.step_height, PC_DARK_BLUE);
 					}
-					DrawString(r.left + WD_FRAMERECT_LEFT, r.right - WD_FRAMERECT_RIGHT, y, item->title, _fios_colours[item->type]);
+					DrawString(r.left + WD_FRAMERECT_LEFT, r.right - WD_FRAMERECT_RIGHT, y + this->resize.step_height / 2, item->title, _fios_colours[item->type]);
 					y += this->resize.step_height;
 					if (y >= this->vscroll->GetCapacity() * this->resize.step_height + r.top + WD_FRAMERECT_TOP) break;
 				}
-- 
2.55.0

