From 5b75bcffbf076fdcf616500531f6b0e84d4c1e2a Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 22 Mar 2014 03:10:48 +0200
Subject: [PATCH 0100/1053] Updated TODO

---
 todo.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/todo.txt b/todo.txt
index ae711782eb..36c93f49af 100644
--- a/todo.txt
+++ b/todo.txt
@@ -1,2 +1 @@
-- Fix NewGRF dialog
 - Add Load button to Play Heightmap dialog
-- 
2.55.0

