From f4612e40316f9fffd551ec055acc5ac560659bef Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 22 Mar 2014 23:51:45 +0200
Subject: [PATCH 0103/1053] Oops x2 - I've copied stuff to the wrong directory

---
 src/fileio.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/fileio.cpp b/src/fileio.cpp
index cb750b614b..707eae5c56 100644
--- a/src/fileio.cpp
+++ b/src/fileio.cpp
@@ -1310,9 +1310,12 @@ void DeterminePaths(const char *exe)
 	char curdir[PATH_MAX];
 	if (getcwd(curdir, sizeof(curdir)) && strstr(curdir, "org.openttd.sdl.lowmem")) {
 		// No, I won't implement file copying in C, shell script is just fine for this job
+		DEBUG(misc, 1, "Copying savegames from ../../org.openttd.sdl/files/.openttd/save to %s", curdir);
 		system("cd ../../org.openttd.sdl/files/.openttd/save && "
-				"for F in *.sav ; do ls \"../../../../org.openttd.sdl/files/.openttd/save/$F\" || "
-				"cat \"$F\" > \"../../../../org.openttd.sdl/files/.openttd/save/$F\" ; done");
+				"for F in *.sav ; do "
+				"ls \"../../../../org.openttd.sdl.lowmem/files/.openttd/save/$F\" || "
+				"cat \"$F\" > \"../../../../org.openttd.sdl.lowmem/files/.openttd/save/$F\" ; "
+				"done");
 		chdir(curdir);
 	}
 #endif
-- 
2.55.0

