From b4cfc84fc7c0e28ae02885a1cd6fc3141ffc385e Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 19 Jul 2014 23:39:41 +0300
Subject: [PATCH 0150/1053] Fixed 'Replace vehicles' window too tall

---
 src/autoreplace_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/autoreplace_gui.cpp b/src/autoreplace_gui.cpp
index 79d8b43712..4310f341a4 100644
--- a/src/autoreplace_gui.cpp
+++ b/src/autoreplace_gui.cpp
@@ -245,7 +245,7 @@ public:
 			case WID_RV_LEFT_MATRIX:
 			case WID_RV_RIGHT_MATRIX:
 				resize->height = GetEngineListHeight((VehicleType)this->window_number);
-				size->height = (this->window_number <= VEH_ROAD ? 8 : 4) * resize->height;
+				size->height = 4 * resize->height;
 				break;
 
 			case WID_RV_LEFT_DETAILS:
-- 
2.55.0

