From 6c9fffd545714e6b2631b8c16e483e4fa1f4fdd9 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 20 Jul 2014 00:16:43 +0300
Subject: [PATCH 0153/1053] Frameskip when pressing fast forward button

---
 src/video/sdl_v.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/video/sdl_v.cpp b/src/video/sdl_v.cpp
index a2ac87dff7..f3ed94d5e8 100644
--- a/src/video/sdl_v.cpp
+++ b/src/video/sdl_v.cpp
@@ -155,6 +155,16 @@ static void DrawSurfaceToScreen()
 {
 	int n = _num_dirty_rects;
 	if (n == 0) return;
+	static int frameskip = 0;
+
+#ifdef __ANDROID__
+	if (_fast_forward) {
+		frameskip++;
+		if (frameskip < 5)
+			return;
+		frameskip = 0;
+	}
+#endif
 
 	_num_dirty_rects = 0;
 	if (n > MAX_DIRTY_RECTS) {
-- 
2.55.0

