From 5934a69f679ad732336487f478e1187273371365 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Wed, 12 Nov 2014 01:15:33 +0200
Subject: [PATCH 0167/1053] Fixed played time display for cloud saves

---
 src/openttd.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/openttd.cpp b/src/openttd.cpp
index bc5b8cbb82..b7a58b041b 100644
--- a/src/openttd.cpp
+++ b/src/openttd.cpp
@@ -1177,7 +1177,8 @@ void SwitchToMode(SwitchMode new_mode)
 					}
 					MakeScreenshot(SC_VIEWPORT, NETWORK_SAVE_SCREENSHOT_FILE);
 					FioFindFullPath(screenshotFile, sizeof(screenshotFile), SCREENSHOT_DIR, NETWORK_SAVE_SCREENSHOT_FILE_PNG);
-					int ret = SDL_ANDROID_CloudSave(_file_to_saveload.name, lastPart, "OpenTTD", lastPart, screenshotFile, _date);
+					uint64_t playedTime = abs(_date - DAYS_TILL(_settings_newgame.game_creation.starting_year)) * 1000;
+					int ret = SDL_ANDROID_CloudSave(_file_to_saveload.name, lastPart, "OpenTTD", lastPart, screenshotFile, playedTime);
 					if (_settings_client.gui.save_to_network == 2) {
 						_settings_client.gui.save_to_network = ret ? 1 : 0;
 					}
-- 
2.55.0

