From 9875c976616f4d7b32e96304ecc625dc57e5a5a0 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 16 Nov 2014 01:59:32 +0200
Subject: [PATCH 0175/1053] Do not change windows focus on mouse-up action

---
 src/window.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/window.cpp b/src/window.cpp
index ffd620b282..e965f7d321 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -2894,7 +2894,7 @@ static void MouseLoop(MouseClick click, int mousewheel)
 
 	if (w == NULL) return;
 
-	if (click != MC_NONE && click != MC_HOVER && !MaybeBringWindowToFront(w)) return;
+	if (click != MC_NONE && click != MC_HOVER && click != MC_LEFT_UP && !MaybeBringWindowToFront(w)) return;
 
 	if (mousewheel != 0) {
 		/* Send mousewheel event to window */
-- 
2.55.0

