From d9f14a5b790c87b036f77736f55533bd15d4a9c5 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 16 Nov 2014 03:53:59 +0200
Subject: [PATCH 0177/1053] Hide all windows when placing airport

---
 src/airport_gui.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/airport_gui.cpp b/src/airport_gui.cpp
index a6afc641eb..aaa7188071 100644
--- a/src/airport_gui.cpp
+++ b/src/airport_gui.cpp
@@ -88,7 +88,7 @@ struct BuildAirToolbarWindow : Window {
 	{
 		switch (widget) {
 			case WID_AT_AIRPORT:
-				if (HandlePlacePushButton(this, WID_AT_AIRPORT, SPR_CURSOR_AIRPORT, HT_RECT | HT_SCROLL_VIEWPORT)) {
+				if (HandlePlacePushButton(this, WID_AT_AIRPORT, SPR_CURSOR_AIRPORT, HT_RECT)) {
 					ShowBuildAirportPicker(this);
 					this->last_user_action = widget;
 				}
@@ -108,6 +108,7 @@ struct BuildAirToolbarWindow : Window {
 		switch (this->last_user_action) {
 			case WID_AT_AIRPORT: {
 				VpStartPlaceSizing(tile, VPM_SINGLE_TILE, DDSP_BUILD_STATION);
+				MoveAllWindowsOffScreen();
 				break;
 			}
 
@@ -130,6 +131,7 @@ struct BuildAirToolbarWindow : Window {
 		switch (select_proc) {
 			case DDSP_BUILD_STATION:
 				assert(start_tile == end_tile);
+				MoveAllHiddenWindowsBackToScreen();
 				PlaceAirport(end_tile);
 				break;
 			case DDSP_DEMOLISH_AREA:
-- 
2.55.0

