From d260038aac12353d5718a9d5944b57f1281a8b13 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 16 Nov 2014 17:12:12 +0200
Subject: [PATCH 0179/1053] Fixed road tunnel preview not shown when dragging
 mouse

---
 src/road_gui.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/road_gui.cpp b/src/road_gui.cpp
index 9cf52681fa..9399b066b1 100644
--- a/src/road_gui.cpp
+++ b/src/road_gui.cpp
@@ -548,6 +548,11 @@ struct BuildRoadToolbarWindow : Window {
 
 	virtual void OnPlaceDrag(ViewportPlaceMethod select_method, ViewportDragDropSelectionProcess select_proc, Point pt)
 	{
+		if (this->last_started_action == WID_ROT_BUILD_TUNNEL) {
+			this->OnPlacePresize(pt, TileVirtXY(pt.x, pt.y));
+			return;
+		}
+
 		/* Here we update the end tile flags
 		 * of the road placement actions.
 		 * At first we reset the end halfroad
-- 
2.55.0

