From dce1a08117dbebd0402d595af28c0ab5bc6b86c3 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Tue, 10 Mar 2015 23:34:53 +0200
Subject: [PATCH 0181/1053] Oops

---
 src/window.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/window.cpp b/src/window.cpp
index 93a4736855..4f078611d1 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -1538,6 +1538,7 @@ static bool IsGoodAutoPlace1(int left, int top, int width, int height, Point &po
 	int bottom = height + top;
 
 	const Window *main_toolbar = FindWindowByClass(WC_MAIN_TOOLBAR);
+	bool vertical_toolbar = _settings_client.gui.vertical_toolbar && _game_mode != GM_EDITOR;
 	if (!vertical_toolbar || !main_toolbar) {
 		if (left < 0 || (main_toolbar != NULL && top < main_toolbar->height) || right > _screen.width || bottom > _screen.height) return false;
 	} else {
-- 
2.55.0

