From 09b117c82460d22fa3106582cf2e3146759a583b Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 12 Apr 2015 21:56:00 +0300
Subject: [PATCH 0192/1053] Added BLACK_REMAP mode in 16bpp blitter

---
 src/blitter/16bpp_simple.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/blitter/16bpp_simple.cpp b/src/blitter/16bpp_simple.cpp
index 675b58b87d..fc0e83fff3 100644
--- a/src/blitter/16bpp_simple.cpp
+++ b/src/blitter/16bpp_simple.cpp
@@ -66,6 +66,12 @@ void Blitter_16bppSimple::Draw(const Blitter::BlitterParams *bp, ZoomLevel zoom)
 					if (src->a != 0) *dst = MakeTransparent(*dst, 192);
 					break;
 
+				case BM_BLACK_REMAP:
+					if (src->a != 0) {
+						*dst = Colour16(0, 0, 0);
+					}
+					break;
+
 				default:
 					if (src->a != 0) *dst = ComposeColourPA(src->c, src->a, *dst);
 					break;
@@ -84,6 +90,7 @@ void Blitter_16bppSimple::Draw(Blitter::BlitterParams *bp, BlitterMode mode, Zoo
 		case BM_COLOUR_REMAP: Draw<BM_COLOUR_REMAP>(bp, zoom); return;
 		case BM_TRANSPARENT:  Draw<BM_TRANSPARENT> (bp, zoom); return;
 		case BM_CRASH_REMAP:  Draw<BM_CRASH_REMAP> (bp, zoom); return;
+		case BM_BLACK_REMAP:  Draw<BM_BLACK_REMAP> (bp, zoom); return;
 	}
 }
 
-- 
2.55.0

