From 6910735abf88c92562171d01516f3ae4de7a84dc Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Mon, 13 Apr 2015 01:25:26 +0300
Subject: [PATCH 0203/1053] Fixed OpenTTD release version, for the third time
 eh

---
 findversion.sh | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/findversion.sh b/findversion.sh
index cca88d51b0..bcae7b1989 100755
--- a/findversion.sh
+++ b/findversion.sh
@@ -70,7 +70,11 @@ ROOT_DIR=`pwd`
 # Determine if we are using a modified version
 # Assume the dir is not modified
 MODIFIED="0"
-if [ -d "$ROOT_DIR/.svn" ] || [ -d "$ROOT_DIR/../.svn" ]; then
+if [ -f "$ROOT_DIR/.ottdrev" ]; then
+	# We are an exported source bundle
+	cat $ROOT_DIR/.ottdrev
+	exit
+elif [ -d "$ROOT_DIR/.svn" ] || [ -d "$ROOT_DIR/../.svn" ]; then
 	# We are an svn checkout
 	if [ -n "`svnversion | grep 'M'`" ]; then
 		MODIFIED="2"
@@ -122,10 +126,6 @@ elif [ -d "$ROOT_DIR/.hg" ]; then
 		# No rev? Maybe it is a custom hgsubversion clone
 		REV_NR=`LC_ALL=C HGPLAIN= hg parent --template="{svnrev}"`
 	fi
-elif [ -f "$ROOT_DIR/.ottdrev" ]; then
-	# We are an exported source bundle
-	cat $ROOT_DIR/.ottdrev
-	exit
 else
 	# We don't know
 	MODIFIED="1"
-- 
2.55.0

