From c7087a15b77c599d3c62dbe97a9915e7f94bc09a Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Tue, 14 Apr 2015 13:05:14 +0300
Subject: [PATCH 0207/1053] Fixed 'new game' dialog

---
 src/genworld_gui.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/genworld_gui.cpp b/src/genworld_gui.cpp
index bc40434df5..5549be9bb9 100644
--- a/src/genworld_gui.cpp
+++ b/src/genworld_gui.cpp
@@ -28,6 +28,7 @@
 #include "core/random_func.hpp"
 #include "progress.h"
 #include "error.h"
+#include "settings_gui.h"
 
 #include "widgets/genworld_widget.h"
 
@@ -468,7 +469,8 @@ struct GenerateLandscapeWindow : public Window {
 			case WID_GL_HEIGHTMAP_SIZE_TEXT:
 				SetDParam(0, this->x);
 				SetDParam(1, this->y);
-				*size = maxdim(*size, GetStringBoundingBox(STR_MAPGEN_HEIGHTMAP_SIZE));
+				//*size = maxdim(*size, GetStringBoundingBox(STR_MAPGEN_HEIGHTMAP_SIZE));
+				size->height = SETTING_BUTTON_HEIGHT;
 				break;
 
 			case WID_GL_TOWN_PULLDOWN:
@@ -492,18 +494,19 @@ struct GenerateLandscapeWindow : public Window {
 			case WID_GL_VARIETY_PULLDOWN:    strs = _variety; break;
 			case WID_GL_HEIGHTMAP_ROTATION_PULLDOWN: strs = _rotation; break;
 			case WID_GL_BORDERS_RANDOM:
-				*size = maxdim(GetStringBoundingBox(STR_MAPGEN_BORDER_RANDOMIZE), GetStringBoundingBox(STR_MAPGEN_BORDER_MANUAL));
+				//*size = maxdim(GetStringBoundingBox(STR_MAPGEN_BORDER_RANDOMIZE), GetStringBoundingBox(STR_MAPGEN_BORDER_MANUAL));
 				break;
 
 			case WID_GL_WATER_NE:
 			case WID_GL_WATER_NW:
 			case WID_GL_WATER_SE:
 			case WID_GL_WATER_SW:
-				*size = maxdim(GetStringBoundingBox(STR_MAPGEN_BORDER_RANDOM), maxdim(GetStringBoundingBox(STR_MAPGEN_BORDER_WATER), GetStringBoundingBox(STR_MAPGEN_BORDER_FREEFORM)));
+				//*size = maxdim(GetStringBoundingBox(STR_MAPGEN_BORDER_RANDOM), maxdim(GetStringBoundingBox(STR_MAPGEN_BORDER_WATER), GetStringBoundingBox(STR_MAPGEN_BORDER_FREEFORM)));
 				break;
 
 			case WID_GL_HEIGHTMAP_NAME_TEXT:
 				size->width = 0;
+				size->height = SETTING_BUTTON_HEIGHT;
 				break;
 
 			default:
-- 
2.55.0

