From 1ea2ca0482421b882e649856066de5285bdaafd0 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Tue, 28 Apr 2015 23:31:53 +0300
Subject: [PATCH 0213/1053] Adjusting vehicle window size

---
 src/group_gui.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/group_gui.cpp b/src/group_gui.cpp
index bff61cb552..9b71650ba6 100644
--- a/src/group_gui.cpp
+++ b/src/group_gui.cpp
@@ -369,7 +369,7 @@ public:
 				resize->height = this->tiny_step_height;
 
 				/* Minimum height is the height of the list widget minus all and default vehicles... */
-				size->height = (this->vli.vtype >= VEH_SHIP ? 4 : 7) * GetVehicleListHeight(this->vli.vtype, this->tiny_step_height) - 2 * this->tiny_step_height;
+				size->height = (this->vli.vtype >= VEH_SHIP ? 3.5 : 7) * GetVehicleListHeight(this->vli.vtype, this->tiny_step_height) - 2 * this->tiny_step_height;
 
 				/* Get a multiple of tiny_step_height of that amount */
 				size->height = Ceil(size->height, tiny_step_height);
@@ -393,7 +393,7 @@ public:
 			case WID_GL_LIST_VEHICLE:
 				this->ComputeGroupInfoSize();
 				resize->height = GetVehicleListHeight(this->vli.vtype, this->tiny_step_height);
-				size->height = (this->vli.vtype >= VEH_SHIP ? 4 : 6) * resize->height;
+				size->height = (this->vli.vtype >= VEH_SHIP ? 3 : 6) * resize->height;
 				break;
 
 			case WID_GL_MANAGE_VEHICLES_DROPDOWN: {
-- 
2.55.0

