From cfa1c18469bb24506efd4820aab9a57ec6791827 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Mon, 4 Apr 2016 23:11:37 +0300
Subject: [PATCH 0250/1053] Fixed bridge UI

---
 src/bridge_gui.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bridge_gui.cpp b/src/bridge_gui.cpp
index 3c6ad22016..ad00fba764 100644
--- a/src/bridge_gui.cpp
+++ b/src/bridge_gui.cpp
@@ -199,6 +199,7 @@ public:
 				sprite_dim.height++; // Sprite is rendered one pixel down in the matrix field.
 				text_dim.height++; // Allowing the bottom row pixels to be rendered on the edge of the matrix field.
 				resize->height = max(sprite_dim.height, text_dim.height) + 2; // Max of both sizes + account for matrix edges.
+				resize->height = GetMinSizing(NWST_STEP, resize->height);
 
 				this->bridgetext_offset = WD_MATRIX_LEFT + sprite_dim.width + 1; // Left edge of text, 1 pixel distance from the sprite.
 				size->width = this->bridgetext_offset + text_dim.width + WD_MATRIX_RIGHT;
-- 
2.55.0

