From a1a662782c10fbb17b2bb72a7a71f93b9ac127f2 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 6 Apr 2016 23:39:17 +0300
Subject: [PATCH 0253/1053] Keep building area highlighted when confirmation
 window is shown

---
 src/viewport.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/viewport.cpp b/src/viewport.cpp
index 251bdfc0a4..ab5555d31e 100644
--- a/src/viewport.cpp
+++ b/src/viewport.cpp
@@ -2449,6 +2449,8 @@ void UpdateTileSelection()
 		}
 	}
 
+	if (ConfirmationWindowShown()) return;
+
 	/* redraw selection */
 	if (_thd.drawstyle != new_drawstyle ||
 			_thd.pos.x != _thd.new_pos.x || _thd.pos.y != _thd.new_pos.y ||
-- 
2.55.0

