From e006df1226c1eb1565ddf39b222447718377eeb9 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Fri, 8 Apr 2016 17:18:23 +0300
Subject: [PATCH 0256/1053] Reverted some unnecessary changes

---
 src/road_gui.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/road_gui.cpp b/src/road_gui.cpp
index 6ddc1ba005..48102ecbff 100644
--- a/src/road_gui.cpp
+++ b/src/road_gui.cpp
@@ -1039,10 +1039,6 @@ struct BuildRoadStationWindow : public PickerWindowBase {
 	{
 		if (!IsInsideMM(widget, WID_BROS_STATION_NE, WID_BROS_STATION_Y + 1)) return;
 
-		int x = Center(r.left + TILE_PIXELS, r.right - r.left, 2 * TILE_PIXELS);
-		/* Height of bus/truck sprite in OpenGFX is TILE_PIXELS + 11. */
-		int y = Center(r.top + WD_FRAMERECT_TOP - WD_MATRIX_BOTTOM + IsWidgetLowered(widget) + 11, r.bottom - r.top, TILE_PIXELS + 11);
-
 		StationType st = (this->window_class == WC_BUS_STATION) ? STATION_BUS : STATION_TRUCK;
 		StationPickerDrawSprite(r.left + 1 + ScaleGUITrad(31), r.bottom - ScaleGUITrad(31), st, INVALID_RAILTYPE, widget < WID_BROS_STATION_X ? ROADTYPE_ROAD : _cur_roadtype, widget - WID_BROS_STATION_NE);
 	}
-- 
2.55.0

