From 9cf2149c7813974e0a3ed7ca24041935611c8176 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Mon, 11 Apr 2016 00:48:14 +0300
Subject: [PATCH 0264/1053] Fixed UI when building with non-autorail tool with
 confirmation window

---
 src/viewport.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/viewport.cpp b/src/viewport.cpp
index 5cd2cb155a..3e7e41bfb1 100644
--- a/src/viewport.cpp
+++ b/src/viewport.cpp
@@ -2438,8 +2438,10 @@ void UpdateTileSelection()
 
 						default: NOT_REACHED();
 					}
-					_thd.selstart.x = x1 & ~TILE_UNIT_MASK;
-					_thd.selstart.y = y1 & ~TILE_UNIT_MASK;
+					if (!ConfirmationWindowShown()) {
+						_thd.selstart.x = x1 & ~TILE_UNIT_MASK;
+						_thd.selstart.y = y1 & ~TILE_UNIT_MASK;
+					}
 					break;
 				default:
 					NOT_REACHED();
-- 
2.55.0

