From 6c6c78752a2a420b78fb5952412b2d4f53bc2a6c Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Tue, 12 Apr 2016 21:16:21 +0300
Subject: [PATCH 0269/1053] Fixed crash when selecting different font size

---
 src/settings_gui.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/settings_gui.cpp b/src/settings_gui.cpp
index 0ec8e9491b..d0d697c1ed 100644
--- a/src/settings_gui.cpp
+++ b/src/settings_gui.cpp
@@ -572,6 +572,7 @@ struct GameOptionsWindow : Window {
 				_freetype.large.size = _freetype.medium.size * 16 / 12;
 				_freetype.mono.size = _freetype.medium.size;
 				ReconstructUserInterface();
+				break;
 
 			case WID_GO_GUI_ZOOM_DROPDOWN:
 				GfxClearSpriteCache();
-- 
2.55.0

