From 4909bdbecc6cf8741ebb2e460f5ff06a50f46ee2 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 16 Apr 2016 22:47:13 +0300
Subject: [PATCH 0272/1053] Fixed joining rail stations broken

---
 src/rail_gui.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/rail_gui.cpp b/src/rail_gui.cpp
index 37000b162e..10869fbbd2 100644
--- a/src/rail_gui.cpp
+++ b/src/rail_gui.cpp
@@ -280,12 +280,15 @@ static bool RailToolbar_CtrlChanged(Window *w)
 	if (w->IsWidgetDisabled(WID_RAT_REMOVE)) return false;
 
 	/* allow ctrl to switch remove mode only for these widgets */
+	// This breaks joining rail stations functionality on Android
+	/*
 	for (uint i = WID_RAT_BUILD_NS; i <= WID_RAT_BUILD_STATION; i++) {
 		if ((i <= WID_RAT_AUTORAIL || i >= WID_RAT_BUILD_WAYPOINT) && w->IsWidgetLowered(i)) {
 			ToggleRailButton_Remove(w);
 			return true;
 		}
 	}
+	*/
 
 	return false;
 }
-- 
2.55.0

