From 21c13781e466366584bcc1542b8192eb7d6c4e72 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Mon, 16 May 2016 22:29:57 +0300
Subject: [PATCH 0281/1053] Fixed a crash in confirmation dialog

---
 src/viewport.cpp | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/src/viewport.cpp b/src/viewport.cpp
index 72e30f1347..e404b8db09 100644
--- a/src/viewport.cpp
+++ b/src/viewport.cpp
@@ -3174,17 +3174,6 @@ place_mouseup:
 	return ES_HANDLED;
 }
 
-void ConfirmPlacingObject()
-{
-	Window *w = _thd.GetCallbackWnd();
-	if (w == NULL) {
-		ResetObjectToPlace();
-		return;
-	}
-
-	w->OnPlaceMouseUp(_thd.select_method, _thd.select_proc, _thd.selend, TileVirtXY(_thd.selstart.x, _thd.selstart.y), TileVirtXY(_thd.selend.x, _thd.selend.y));
-}
-
 /**
  * Change the cursor and mouse click/drag handling to a mode for performing special operations like tile area selection, object placement, etc.
  * @param icon New shape of the mouse cursor.
@@ -3202,6 +3191,20 @@ void SetObjectToPlaceWnd(CursorID icon, PaletteID pal, HighLightStyle mode, Wind
 static WindowClass _last_selected_window_class;
 static WindowNumber _last_selected_window_number;
 
+/** Place object from the build confirmation dialog */
+void ConfirmPlacingObject()
+{
+	Window *w = _thd.GetCallbackWnd();
+	if (w == NULL) {
+		ResetObjectToPlace();
+		return;
+	}
+
+	_last_selected_window_class = _thd.window_class;
+	_last_selected_window_number = _thd.window_number;
+	w->OnPlaceMouseUp(_thd.select_method, _thd.select_proc, _thd.selend, TileVirtXY(_thd.selstart.x, _thd.selstart.y), TileVirtXY(_thd.selend.x, _thd.selend.y));
+}
+
 /**
  * Change the cursor and mouse click/drag handling to a mode for performing special operations like tile area selection, object placement, etc.
  * @param icon New shape of the mouse cursor.
-- 
2.55.0

