From c9b0eea04ced67fac9ba097b5b64073b9a3469c9 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Fri, 1 Jul 2016 22:57:36 +0300
Subject: [PATCH 0295/1053] Smallmap: disabled show/hide legend button

---
 src/smallmap_gui.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/smallmap_gui.cpp b/src/smallmap_gui.cpp
index 3eacb7f902..812d1b243f 100644
--- a/src/smallmap_gui.cpp
+++ b/src/smallmap_gui.cpp
@@ -1063,7 +1063,7 @@ SmallMapWindow::SmallMapWindow(WindowDesc *desc, int window_number) :
 	//this->SetWidgetLoweredState(WID_SM_SHOW_HEIGHT, _smallmap_show_heightmap);
 
 	this->SetWidgetLoweredState(WID_SM_TOGGLETOWNNAME, this->show_towns);
-	this->SetWidgetLoweredState(WID_SM_SHOW_LEGEND, this->show_legend);
+	//this->SetWidgetLoweredState(WID_SM_SHOW_LEGEND, this->show_legend);
 
 	this->SetupWidgetData();
 
@@ -1770,8 +1770,8 @@ static const NWidgetPart _nested_smallmap_bar[] = {
 			NWidget(NWID_HORIZONTAL),
 				/* Top button row. */
 				NWidget(NWID_HORIZONTAL, NC_EQUALSIZE),
-					NWidget(WWT_IMGBTN, COLOUR_BROWN, WID_SM_SHOW_LEGEND),
-							SetDataTip(SPR_IMG_QUERY, STR_SMALLMAP_TOOLTIP_SHOW_LEGEND), SetFill(1, 1),
+					//NWidget(WWT_IMGBTN, COLOUR_BROWN, WID_SM_SHOW_LEGEND),
+					//		SetDataTip(SPR_IMG_QUERY, STR_SMALLMAP_TOOLTIP_SHOW_LEGEND), SetFill(1, 1),
 					NWidget(WWT_PUSHIMGBTN, COLOUR_BROWN, WID_SM_ZOOM_IN),
 							SetDataTip(SPR_IMG_ZOOMIN, STR_TOOLBAR_TOOLTIP_ZOOM_THE_VIEW_IN), SetFill(1, 1),
 					NWidget(WWT_PUSHIMGBTN, COLOUR_BROWN, WID_SM_ZOOM_OUT),
-- 
2.55.0

