From 11eb95aae9923a0e69179b3f6c3256f8e9ec5eb9 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Fri, 13 Jan 2017 15:15:22 +0200
Subject: [PATCH 0300/1053] Code style

---
 src/tutorial_gui.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/tutorial_gui.cpp b/src/tutorial_gui.cpp
index 7a7eab80ea..6166831592 100644
--- a/src/tutorial_gui.cpp
+++ b/src/tutorial_gui.cpp
@@ -176,7 +176,7 @@ struct TutorialWindow : public Window {
 		// because youtube app freezes a screen for a second before launching.
 		if (this->counter > 0) {
 			this->counter--;
-			if (counter == 1) {
+			if (this->counter == 1) {
 				this->RaiseWidget(WID_STL_BUS);
 				this->RaiseWidget(WID_STL_TRUCK);
 				this->RaiseWidget(WID_STL_TRAIN);
@@ -185,9 +185,10 @@ struct TutorialWindow : public Window {
 				this->RaiseWidget(WID_STL_FACILALL);
 				this->SetDirty();
 			}
-			if (counter == 0) {
-				if (video) {
-					OpenExternTutorialVideo(video);
+			if (this->counter == 0) {
+				if (this->video) {
+					OpenExternTutorialVideo(this->video);
+					this->video = NULL;
 				}
 			}
 		}
-- 
2.55.0

