From d35d97c4a5d91350b1cef34a8f93db297a9216bf Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 28 Jan 2017 23:03:54 +0200
Subject: [PATCH 0306/1053] Adjusted default video settings for touchscreen

---
 src/table/misc_settings.ini | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/table/misc_settings.ini b/src/table/misc_settings.ini
index 5755356437..15c6eb733f 100644
--- a/src/table/misc_settings.ini
+++ b/src/table/misc_settings.ini
@@ -121,7 +121,7 @@ name     = ""resolution""
 type     = SLE_INT
 length   = 2
 var      = _cur_resolution
-def      = ""640,480""
+def      = ""854,480""
 cat      = SC_BASIC
 
 [SDTG_STR]
@@ -235,7 +235,7 @@ def      = false
 name     = ""min_button_size""
 type     = SLE_UINT
 var      = _settings_client.gui.min_button
-def      = 0
+def      = 40
 min      = 0
 max      = 100
 cat      = SC_EXPERT
@@ -244,7 +244,7 @@ cat      = SC_EXPERT
 name     = ""min_step_size""
 type     = SLE_UINT
 var      = _settings_client.gui.min_step
-def      = 0
+def      = 40
 min      = 0
 max      = 100
 cat      = SC_EXPERT
-- 
2.55.0

