From e16c85f53aa7c2f63652ab69c64e8aa0e55c8382 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 28 Jan 2017 23:28:11 +0200
Subject: [PATCH 0307/1053] Fixed libtimidity not detected on Android

---
 configure   | 2 +-
 source.list | 5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 6b2c9d87b7..180303543b 100755
--- a/configure
+++ b/configure
@@ -127,7 +127,7 @@ AWKCOMMAND='
 		if ($0 == "WINCE"       && "'$os'" != "WINCE")             { next; }
 		if ($0 == "MSVC"        && "'$os'" != "MSVC")              { next; }
 		if ($0 == "DIRECTMUSIC" && "'$with_direct_music'" == "0")  { next; }
-		if ($0 == "LIBTIMIDITY" && "'$libtimidity'" == "" )        { next; }
+		if ($0 == "LIBTIMIDITY" && "'$with_libtimidity'" == "0" )  { next; }
 		if ($0 == "HAVE_THREAD" && "'$with_threads'" == "0")       { next; }
 		if ($0 == "SSE"         && "'$with_sse'" != "1")           { next; }
 
diff --git a/source.list b/source.list
index 890185fc83..f7c8096d57 100644
--- a/source.list
+++ b/source.list
@@ -1123,7 +1123,10 @@ music/null_m.cpp
 			#else
 				#if MORPHOS
 				#else
-					music/extmidi.cpp
+					#if LIBTIMIDITY
+					#else
+						music/extmidi.cpp
+					#end
 				#end
 			#end
 		#end
-- 
2.55.0

