From f87f657e4dbd82a24f5887e7d21a0fe98a4e1db6 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 29 Jan 2017 15:57:47 +0200
Subject: [PATCH 0310/1053] Button size goes up to 100

---
 src/settings_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/settings_gui.cpp b/src/settings_gui.cpp
index e4a193a769..2b68f3093a 100644
--- a/src/settings_gui.cpp
+++ b/src/settings_gui.cpp
@@ -47,7 +47,7 @@
 #include <SDL_android.h>
 #endif
 
-enum { MIN_BUTTON_SIZE = 10, MAX_BUTTON_SIZE = 40 };
+enum { MIN_BUTTON_SIZE = 10, MAX_BUTTON_SIZE = 100 };
 
 static const StringID _driveside_dropdown[] = {
 	STR_GAME_OPTIONS_ROAD_VEHICLES_DROPDOWN_LEFT,
-- 
2.55.0

