From 03868523af7ec32c22bb7349a1c325d3fa505bfb Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Thu, 23 Mar 2017 23:12:47 +0200
Subject: [PATCH 0331/1053] Fixed windows resizing with no titlebars

---
 src/widget.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/widget.cpp b/src/widget.cpp
index 7039fa313c..014a0be7ec 100644
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -2242,6 +2242,10 @@ void NWidgetLeaf::SetupSmallestSize(Window *w, bool init_array)
 		size = extra;
 		fill = extra;
 		resize = extra;
+		if (this->type == WWT_CAPTION) {
+			fill.width = 1;
+			resize.width = 1;
+		}
 	} else switch (this->type) {
 		case WWT_EMPTY: {
 			static const Dimension extra = {0, 0};
-- 
2.55.0

