From 7f666c5547495dd6893f181f5b7ed1cce9c46327 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sat, 25 Mar 2017 23:24:10 +0200
Subject: [PATCH 0335/1053] Updated version file

---
 .ottdrev | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.ottdrev b/.ottdrev
index 89af2c77e7..23cb3a5053 100644
--- a/.ottdrev
+++ b/.ottdrev
@@ -1 +1 @@
-1.6.1	27609	0	1.6.1
+1.7.0-RC1	27783	0	1.7.0-RC1
-- 
2.55.0

