From 8afff1ff7978e2de921c5252883a847964110e2f Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Tue, 28 Mar 2017 23:48:36 +0300
Subject: [PATCH 0356/1053] Updated todo

---
 todo.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/todo.txt b/todo.txt
index 1efb950c71..42290f4622 100644
--- a/todo.txt
+++ b/todo.txt
@@ -22,3 +22,5 @@
 - Main menu is too big.
 
 - Cannot close tutorial menu.
+
+- Main menu can be closed permanently.
-- 
2.55.0

