From 63850a3172ea81d1b3c56f6b181120b8790fbdd4 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Wed, 29 Mar 2017 20:34:02 +0300
Subject: [PATCH 0358/1053] Disabled showing tutorial window on the first start

---
 src/tutorial_gui.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/tutorial_gui.cpp b/src/tutorial_gui.cpp
index 3cf107abce..575d63514e 100644
--- a/src/tutorial_gui.cpp
+++ b/src/tutorial_gui.cpp
@@ -208,6 +208,8 @@ void ShowTutorialWindow()
 
 void ShowTutorialWindowOnceAfterInstall()
 {
+	// Close button on tutorial window is gone, so don't show that windows on first run, it's confusing
+#if 0
 	static const char * TUTORIAL_SHOWN_FLAG = ".tutorial-shown-3.flag";
 
 	FILE *ff = fopen(TUTORIAL_SHOWN_FLAG, "r");
@@ -221,4 +223,5 @@ void ShowTutorialWindowOnceAfterInstall()
 	fprintf(ff, "Tutorial shown");
 	fclose(ff);
 	ShowTutorialWindow();
+#endif
 }
-- 
2.55.0

