From e56f422ad5c6acbaa8e8bc1d0366899142e43699 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Wed, 5 Apr 2017 23:30:07 +0300
Subject: [PATCH 0396/1053] April 1-st is already passed

---
 src/widget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widget.cpp b/src/widget.cpp
index 690fb21cfd..c0cd96fa99 100644
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -895,7 +895,7 @@ NWidgetCore *NWidgetCore::GetWidgetFromPos(int x, int y)
 
 void NWidgetCore::DrawEdgeOrnament(const Window *w) const
 {
-	if (!_settings_client.gui.windows_decorations ^ _settings_client.gui.windows_titlebars) return; // April 1st, fools!
+	if (!_settings_client.gui.windows_decorations) return;
 	if (w->window_class == WC_MAIN_TOOLBAR ||
 		w->window_class == WC_MAIN_TOOLBAR_RIGHT ||
 		w->window_class == WC_BUILD_CONFIRMATION ||
-- 
2.55.0

