From f014b84b1a35e20809ca13c53ccdd06018df54dc Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Tue, 27 Jun 2017 22:01:01 +0300
Subject: [PATCH 0429/1053] Fixed classic mobile toolbar

---
 src/toolbar_gui.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/toolbar_gui.cpp b/src/toolbar_gui.cpp
index 643c08bd70..e530d64fe7 100644
--- a/src/toolbar_gui.cpp
+++ b/src/toolbar_gui.cpp
@@ -2369,16 +2369,16 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 		};
 		static const byte arrange_right_classic_noswitch[] = {
 			WID_TN_RAILS,
-			WID_TN_LANDSCAPE,
 			WID_TN_TRAINS,
+			WID_TN_ROADVEHS,
+			WID_TN_SHIPS,
+			WID_TN_AIRCRAFTS,
 			WID_TN_STATIONS,
 			WID_TN_FINANCES,
 			WID_TN_COMPANIES,
 			WID_TN_GRAPHS,
 			WID_TN_LEAGUE,
 			WID_TN_STATIONS,
-			WID_TN_MUSIC_SOUND,
-			WID_TN_MESSAGES,
 			WID_TN_HELP,
 		};
 
-- 
2.55.0

