From 97b73144ffdd1d80d6df3756929a0f2d5492ba50 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 20 Sep 2017 21:34:42 +0300
Subject: [PATCH 0434/1053] Hide system mouse pointer in Android Nougat and
 higher

---
 src/video/sdl_v.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/video/sdl_v.cpp b/src/video/sdl_v.cpp
index e6fbe64d03..b7dee86647 100644
--- a/src/video/sdl_v.cpp
+++ b/src/video/sdl_v.cpp
@@ -30,6 +30,7 @@
 #include <SDL.h>
 #ifdef __ANDROID__
 #include <SDL_screenkeyboard.h>
+#include <SDL_android.h>
 #endif
 
 #include "../safeguards.h"
@@ -429,6 +430,7 @@ bool VideoDriver_SDL::CreateMainSurface(uint w, uint h)
 	if (!_multitouch_device) {
 		_multitouch_device = SDL_JoystickOpen(0);
 	}
+	SDL_ANDROID_SetSystemMousePointerVisible(0); // We have our own cursor, only works on Android N
 #endif
 
 	return true;
-- 
2.55.0

