From 4e2f347a908082b96df01fc83f424dd67f3100e0 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Tue, 26 Dec 2017 21:59:22 +0200
Subject: [PATCH 0446/1053] Fixed pinch-zoom, again

---
 src/video/sdl_v.cpp | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/src/video/sdl_v.cpp b/src/video/sdl_v.cpp
index e604fc5d10..8af376e814 100644
--- a/src/video/sdl_v.cpp
+++ b/src/video/sdl_v.cpp
@@ -588,8 +588,21 @@ int VideoDriver_SDL::PollEvent()
 #endif
 					break;
 
+#ifdef __ANDROID__
+				case SDL_BUTTON_WHEELUP:
+				case SDL_BUTTON_WHEELDOWN:
+					_cursor.wheel += (ev.button.button == SDL_BUTTON_WHEELDOWN) ? 1 : -1;
+					_right_button_down = false;
+					// Center the mouse cursor between touch points
+					SDL_GetMouseState(&_cursor.pos.x, &_cursor.pos.y);
+					_cursor.pos.x = (_cursor.pos.x + _multitouch_second_point.x) / 2;
+					_cursor.pos.y = (_cursor.pos.y + _multitouch_second_point.y) / 2;
+					//_cursor.UpdateCursorPosition(_cursor.pos.x, _cursor.pos.y, false);
+					break;
+#else
 				case SDL_BUTTON_WHEELUP:   _cursor.wheel--; break;
 				case SDL_BUTTON_WHEELDOWN: _cursor.wheel++; break;
+#endif
 
 				default: break;
 			}
@@ -648,14 +661,6 @@ int VideoDriver_SDL::PollEvent()
 				if (ev.key.keysym.sym == SDLK_LSHIFT || ev.key.keysym.sym == SDLK_RSHIFT) {
 					_shift_pressed = true;
 				}
-				if (ev.key.keysym.sym == SDLK_KP_PLUS || ev.key.keysym.sym == SDLK_KP_MINUS) {
-					// Center the mouse cursor between touch points
-					SDL_GetMouseState(&_right_button_down_pos.x, &_right_button_down_pos.y);
-					_right_button_down_pos.x = (_right_button_down_pos.x + _multitouch_second_point.x) / 2;
-					_right_button_down_pos.y = (_right_button_down_pos.y + _multitouch_second_point.y) / 2;
-					_cursor.pos = _right_button_down_pos;
-					//_cursor.UpdateCursorPosition(_cursor.pos.x, _cursor.pos.y, false);
-				}
 #endif
 			}
 			break;
-- 
2.55.0

