From 174b0da90fbe5b22cb15bdef8b91bfeb467352e9 Mon Sep 17 00:00:00 2001
From: PeterN <peter@fuzzle.org>
Date: Sat, 23 Mar 2019 17:59:19 +0000
Subject: [PATCH 0476/1053] Fix: Filtered file list did not scroll properly.
 (#7402)

---
 src/fios_gui.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/fios_gui.cpp b/src/fios_gui.cpp
index 9911de2133..58242e71e2 100644
--- a/src/fios_gui.cpp
+++ b/src/fios_gui.cpp
@@ -437,12 +437,15 @@ public:
 				GfxFillRect(r.left + 1, r.top + 1, r.right, r.bottom, PC_BLACK);
 
 				uint y = r.top + WD_FRAMERECT_TOP;
-				for (uint pos = this->vscroll->GetPosition(); pos < this->fios_items.Length(); pos++) {
-					if (!this->fios_items_shown[pos]) {
+				uint scroll_pos = this->vscroll->GetPosition();
+				for (uint row = 0; row < this->fios_items.Length(); row++) {
+					if (!this->fios_items_shown[row]) {
 						/* The current item is filtered out : we do not show it */
+						scroll_pos++;
 						continue;
 					}
-					const FiosItem *item = this->fios_items.Get(pos);
+					if (row < scroll_pos) continue;
+					const FiosItem *item = this->fios_items.Get(row);
 
 					if (item == this->selected) {
 						GfxFillRect(r.left + 1, y, r.right, y + this->resize.step_height, PC_DARK_BLUE);
-- 
2.55.0

