From 3405519815a5754f4ead9cab8632ed53eed41755 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 24 Apr 2019 18:03:01 +0300
Subject: [PATCH 0499/1053] Fixed missing enum in music widget

---
 src/widgets/music_widget.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/widgets/music_widget.h b/src/widgets/music_widget.h
index c8e6a98186..cd48dc469a 100644
--- a/src/widgets/music_widget.h
+++ b/src/widgets/music_widget.h
@@ -16,8 +16,10 @@
 enum MusicTrackSelectionWidgets {
 	WID_MTS_CAPTION,    ///< Window caption.
 	WID_MTS_LIST_LEFT,  ///< Left button.
+	WID_MTS_LEFT_SCROLLBAR,  ///< Scrollbar of left list.
 	WID_MTS_PLAYLIST,   ///< Playlist.
 	WID_MTS_LIST_RIGHT, ///< Right button.
+	WID_MTS_RIGHT_SCROLLBAR, ///< Scrollbar of right list.
 	WID_MTS_MUSICSET,   ///< Music set selection.
 	WID_MTS_ALL,        ///< All button.
 	WID_MTS_OLD,        ///< Old button.
-- 
2.55.0

