From d91ada456bca04e16e2c5aee5373a5bdfd378a26 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 24 Apr 2019 20:01:45 +0300
Subject: [PATCH 0502/1053] Updated revision number in .ottrev

---
 .ottdrev | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.ottdrev b/.ottdrev
index fdb7582eca..4bfa84ecdd 100644
--- a/.ottdrev
+++ b/.ottdrev
@@ -1 +1 @@
-1.8.0	28002	0	1.8.0
+1.9.1	20190408	0	b61ef7e0d2e11e6f8fa963c9459e918b24afc8b5	1	1
-- 
2.55.0

