From 108d3365ff0d03120953fb46ef3703742f81c3c8 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 24 Apr 2019 20:02:12 +0300
Subject: [PATCH 0503/1053] Renamed WID_TN_AIRCRAFTS to WID_TN_AIRCRAFT

---
 src/toolbar_gui.cpp | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/src/toolbar_gui.cpp b/src/toolbar_gui.cpp
index 5b68d67fbb..82a3f80ea6 100644
--- a/src/toolbar_gui.cpp
+++ b/src/toolbar_gui.cpp
@@ -1943,7 +1943,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_ZOOM_IN,
 			WID_TN_ZOOM_OUT,
 			WID_TN_RAILS,
@@ -1979,7 +1979,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_RAILS,
 			WID_TN_ROADS,
 			WID_TN_WATER,
@@ -2017,7 +2017,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_RAILS,
 			WID_TN_ROADS,
 			WID_TN_WATER,
@@ -2057,7 +2057,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_RAILS,
 			WID_TN_ROADS,
 			WID_TN_WATER,
@@ -2122,7 +2122,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_MUSIC_SOUND,
 			WID_TN_MESSAGES,
 			WID_TN_HELP,
@@ -2142,7 +2142,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_RAILS,
 			WID_TN_ROADS,
 			WID_TN_WATER,
@@ -2187,7 +2187,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_RAILS,
 			WID_TN_ROADS,
 			WID_TN_WATER,
@@ -2243,7 +2243,7 @@ class NWidgetMainToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_ZOOM_IN,
 			WID_TN_ZOOM_OUT,
 			WID_TN_RAILS,
@@ -2320,7 +2320,7 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_GRAPHS,
 			WID_TN_INDUSTRIES,
 			WID_TN_MUSIC_SOUND,
@@ -2332,7 +2332,7 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_FINANCES,
 			WID_TN_COMPANIES,
 			WID_TN_TOWNS,
@@ -2373,7 +2373,7 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_LEAGUE,
 			WID_TN_STATIONS,
 			WID_TN_STORY,
@@ -2387,7 +2387,7 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_STATIONS,
 			WID_TN_FINANCES,
 			WID_TN_COMPANIES,
@@ -2423,7 +2423,7 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_LANDSCAPE,
 			WID_TN_GRAPHS,
 			WID_TN_INDUSTRIES,
@@ -2459,7 +2459,7 @@ class NWidgetVerticalToolbarContainer : public NWidgetToolbarContainer {
 			WID_TN_TRAINS,
 			WID_TN_ROADVEHS,
 			WID_TN_SHIPS,
-			WID_TN_AIRCRAFTS,
+			WID_TN_AIRCRAFT,
 			WID_TN_LANDSCAPE,
 			WID_TN_GOAL,
 			WID_TN_GRAPHS,
-- 
2.55.0

