From ef6378088f79301ddba1e3185d0a86875c3869ec Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 25 Apr 2019 19:08:16 +0300
Subject: [PATCH 0507/1053] Increased video resolutions list size to 100

---
 src/gfx_func.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gfx_func.h b/src/gfx_func.h
index 4486874462..95ba2d3cd1 100644
--- a/src/gfx_func.h
+++ b/src/gfx_func.h
@@ -71,7 +71,7 @@ extern DrawPixelInfo _screen;
 extern bool _screen_disable_anim;   ///< Disable palette animation (important for 32bpp-anim blitter during giant screenshot)
 
 extern int _num_resolutions;
-extern Dimension _resolutions[32];
+extern Dimension _resolutions[100];
 extern Dimension _cur_resolution;
 extern Palette _cur_palette; ///< Current palette
 
-- 
2.55.0

