From fe73e6d021a5f9e4cc851462988d867ae78aca8e Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 11 Jul 2019 23:11:12 +0300
Subject: [PATCH 0519/1053] Removed debug

---
 src/smallmap_gui.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/smallmap_gui.h b/src/smallmap_gui.h
index 9bc072642d..d81c5c77ed 100644
--- a/src/smallmap_gui.h
+++ b/src/smallmap_gui.h
@@ -178,7 +178,6 @@ public:
 	 */
 	inline int GetMinLegendWidth() const
 	{
-		printf("GetMinLegendWidth: this->min_number_of_columns %d this->column_width %d\n", this->min_number_of_columns, this->column_width);
 		return WD_FRAMERECT_LEFT + this->min_number_of_columns * this->column_width;
 	}
 
-- 
2.55.0

