From 5ba3995edb71bfe239c82a6e34580bf80e939a44 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Sat, 31 Aug 2019 19:31:38 +0300
Subject: [PATCH 0525/1053] Cleaned up ugly Android-specifc hack

---
 src/openttd.cpp | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/openttd.cpp b/src/openttd.cpp
index cf2172c464..3d5e51b5a6 100644
--- a/src/openttd.cpp
+++ b/src/openttd.cpp
@@ -500,24 +500,6 @@ struct AfterNewGRFScan : NewGRFScanCallback {
 		}
 #endif /* ENABLE_NETWORK */
 
-		// TODO: remove this hack in one year
-		// Check if OpenTTD config is broken by blitter changing code
-		static bool checked = false;
-		DEBUG(misc, 0, "========= Limits for vehicles: %d %d %d %d", _settings_newgame.vehicle.max_trains, _settings_newgame.vehicle.max_roadveh, _settings_newgame.vehicle.max_aircraft, _settings_newgame.vehicle.max_ships);
-		if (!checked &&
-			_settings_newgame.vehicle.max_trains == 0 && _settings_newgame.vehicle.max_roadveh == 0 &&
-			_settings_newgame.vehicle.max_aircraft == 0 && _settings_newgame.vehicle.max_ships == 0) {
-			if (access(".openttd/openttd-backup.cfg", F_OK) == 0) {
-				system("mv -f .openttd/openttd-backup.cfg .openttd/openttd.cfg");
-			} else {
-				unlink("libsdl-DownloadFinished-1.flag");
-			}
-			//_exit_game = true;
-			//_restart_game = true;
-			_exit(0); // kill(getpid(), SIG_KILL); //abort(); // Kill myself with contempt, user will restart the app, because otherwise we enter infinite restart loop
-		}
-		checked = true;
-
 		/* After the scan we're not used anymore. */
 		delete this;
 	}
-- 
2.55.0

