From 070f1ddf4fa15ffa0aaf267fb72dac55c35d19bf Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Mon, 2 Sep 2019 22:50:26 +0300
Subject: [PATCH 0529/1053] Fixed dropdown list missing items

---
 src/widgets/dropdown.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/widgets/dropdown.cpp b/src/widgets/dropdown.cpp
index 6fd1a4b725..366f146119 100644
--- a/src/widgets/dropdown.cpp
+++ b/src/widgets/dropdown.cpp
@@ -154,6 +154,9 @@ struct DropdownWindow : Window {
 		/* Capacity is the average number of items visible */
 		this->vscroll->SetCapacity(size.height * (uint16)list->Length() / list_height);
 		this->vscroll->SetCount((uint16)list->Length());
+		if (this->vscroll->GetCount() <= this->vscroll->GetCapacity()) {
+			scroll_pos = 0;
+		}
 		this->vscroll->SetPosition(scroll_pos);
 		this->vscroll->UpdatePosition(0);
 
-- 
2.55.0

