From 082c4472050c3801c435b4b0f6ae2da5da354e8c Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Fri, 20 Mar 2020 22:16:19 +0200
Subject: [PATCH 0542/1053] Cleanup

---
 src/window.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/window.cpp b/src/window.cpp
index c9db47bc18..161a86bc36 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -1918,8 +1918,6 @@ static Point LocalGetWindowPlacement(const WindowDesc *desc, int16 sm_width, int
 			NOT_REACHED();
 	}
 
-	// try to put it to
-
 	return pt;
 }
 
-- 
2.55.0

