From 8f1957eac757b388b42100c94e48c0fad818ee32 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Fri, 20 Mar 2020 23:00:51 +0200
Subject: [PATCH 0544/1053] Removed debug

---
 src/window.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/window.cpp b/src/window.cpp
index 7e194eaca0..bdaf3b0635 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -3213,7 +3213,6 @@ static void MouseLoop(MouseClick click, int mousewheel)
 
 			case MC_RIGHT:
 				if (!(w->flags & WF_DISABLE_VP_SCROLL)) {
-					printf("RMB scroll start\n");
 					_scrolling_viewport = true;
 					_cursor.fix_at = (_settings_client.gui.scroll_mode == VSM_VIEWPORT_RMB_FIXED ||
 							_settings_client.gui.scroll_mode == VSM_MAP_RMB_FIXED);
-- 
2.55.0

