From a501a8734c031139a7f204d735609e89b6edd57a Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Sun, 22 Mar 2020 21:45:18 +0200
Subject: [PATCH 0547/1053] Fixed todo

---
 todo.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/todo.txt b/todo.txt
index 3d95f7bc15..fa514af9bd 100644
--- a/todo.txt
+++ b/todo.txt
@@ -9,6 +9,4 @@
 
 - World map is too wide.
 
-- Town directory window crashes the game.
-
 - Build rail window is ugly.
-- 
2.55.0

