From e3e0559677f12c90a8b5961f6adb0904517f721a Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Mon, 23 Mar 2020 00:28:15 +0200
Subject: [PATCH 0550/1053] Updated todo

---
 todo.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/todo.txt b/todo.txt
index c900f3d552..8a5be0006b 100644
--- a/todo.txt
+++ b/todo.txt
@@ -7,6 +7,4 @@
 
 - Cloud save is broken.
 
-- World map is too wide.
-
 - Build rail popup list is ugly.
-- 
2.55.0

