From 8178318e642bf8a9bfa4ac55db3cb0b43b330312 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Mon, 24 Aug 2020 23:25:30 +0300
Subject: [PATCH 0556/1053] Fixed train list button size in news window

---
 src/news_gui.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/news_gui.cpp b/src/news_gui.cpp
index d4ca8155df..958e81b3a4 100644
--- a/src/news_gui.cpp
+++ b/src/news_gui.cpp
@@ -387,6 +387,7 @@ struct NewsWindow : Window {
 					Dimension d2 = GetStringBoundingBox(this->GetWidget<NWidgetCore>(WID_N_SHOW_GROUP)->widget_data);
 					d2.height += WD_CAPTIONTEXT_TOP + WD_CAPTIONTEXT_BOTTOM;
 					d2.width += WD_CAPTIONTEXT_LEFT + WD_CAPTIONTEXT_RIGHT;
+					d2.width = max(d2.width, GetMinSizing(NWST_BUTTON, FONT_HEIGHT_NORMAL));
 					*size = d2;
 				} else {
 					/* Hide 'Show group window' button if this news is not about a vehicle. */
-- 
2.55.0

