From a90cd0680151bfebe78df6f014fa5636a2d2ffd0 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Wed, 6 Jan 2021 01:29:16 +0200
Subject: [PATCH 0562/1053] Fixed music on Android

---
 src/sound/sdl_s.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sound/sdl_s.cpp b/src/sound/sdl_s.cpp
index 16ac64da75..d768d3399c 100644
--- a/src/sound/sdl_s.cpp
+++ b/src/sound/sdl_s.cpp
@@ -33,7 +33,7 @@ extern void Android_MidiMixMusic(Sint16 *stream, int len);
 static void CDECL fill_sound_buffer(void *userdata, Uint8 *stream, int len)
 {
 	MxMixSamples(stream, len / 4);
-#if defined(__ANDROID__) && defined(LIBTIMIDITY)
+#if defined(__ANDROID__)
 	Android_MidiMixMusic((Sint16 *)stream, len / 2);
 #endif
 }
-- 
2.55.0

