From 8c363ccc9d17b421f1100a654eb550a7de89f504 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Mon, 25 Jan 2021 01:10:13 +0200
Subject: [PATCH 0563/1053] Fixed slow compilation on Android

---
 CMakeLists.txt | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad72089be9..73a244cb7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -183,10 +183,10 @@ include_directories(${CMAKE_SOURCE_DIR}/src/3rdparty/squirrel/include)
 include(MSVCFilters)
 
 if(ANDROID)
-add_library(openttd SHARED ${GENERATED_SOURCE_FILES})
-target_link_libraries(openttd log z)
+    add_library(openttd SHARED ${GENERATED_SOURCE_FILES})
+    target_link_libraries(openttd log z)
 else()
-add_executable(openttd WIN32 ${GENERATED_SOURCE_FILES})
+    add_executable(openttd WIN32 ${GENERATED_SOURCE_FILES})
 endif()
 
 set_target_properties(openttd PROPERTIES OUTPUT_NAME "${BINARY_NAME}")
@@ -213,6 +213,11 @@ if(IPO_FOUND)
 endif()
 set_target_properties(openttd PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/bin")
 process_compile_flags()
+if(ANDROID)
+    # A bug in Android NDK r22: https://github.com/android/ndk/issues/1418
+    #get_source_file_property(TMP_CFLAGS ${CMAKE_SOURCE_DIR}/src/settings.cpp COMPILE_FLAGS)
+    set_source_files_properties(${CMAKE_SOURCE_DIR}/src/settings.cpp PROPERTIES COMPILE_FLAGS "-O1")
+endif()
 
 include(LinkPackage)
 link_package(PNG TARGET PNG::PNG ENCOURAGED)
-- 
2.55.0

