From 3e6f1924bb7e9893ddb6b0aa5ed3dcb8ee96a272 Mon Sep 17 00:00:00 2001
From: pelya <x.pelya.x@gmail.com>
Date: Thu, 28 Jan 2021 00:59:32 +0200
Subject: [PATCH 0567/1053] Fixed SONAME in Android shared library

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73a244cb7b..88ef2383ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,6 +184,7 @@ include(MSVCFilters)
 
 if(ANDROID)
     add_library(openttd SHARED ${GENERATED_SOURCE_FILES})
+    set(BINARY_NAME application)
     target_link_libraries(openttd log z)
 else()
     add_executable(openttd WIN32 ${GENERATED_SOURCE_FILES})
-- 
2.55.0

