From 53458e55d70ba062f5d670015aee9aad8ed81ac8 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 3 Mar 2021 01:24:43 +0200
Subject: [PATCH 0579/1053] Fixed NDK armeabi-v7a compilation speed

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4036fd0344..3897396459 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -285,9 +285,9 @@ if(ANDROID)
         ${EXPAT_LIBRARY}
     )
     # A bug in Android NDK r22: https://github.com/android/ndk/issues/1418
-    if(ANDROID_ABI EQUAL "armeabi-v7a" OR ANDROID_ABI EQUAL "x86")
+    if(ANDROID_ABI STREQUAL "armeabi-v7a" OR ANDROID_ABI STREQUAL "x86")
         set_source_files_properties(${CMAKE_SOURCE_DIR}/src/settings.cpp PROPERTIES COMPILE_FLAGS "-O0")
-    endif(ANDROID_ABI)
+    endif()
 endif(ANDROID)
 
 if(APPLE)
-- 
2.55.0

