From b69c7bd928d8066061728ea28238870e7a567aa3 Mon Sep 17 00:00:00 2001
From: dP <dp@dpointer.org>
Date: Sat, 2 Oct 2021 17:33:28 +0300
Subject: [PATCH 0596/1053] Cleanup: Swap comments for IConsoleModes (#9587)

---
 src/console_type.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/console_type.h b/src/console_type.h
index add22e0084..8c40b4f5b5 100644
--- a/src/console_type.h
+++ b/src/console_type.h
@@ -14,9 +14,9 @@
 
 /** Modes of the in-game console. */
 enum IConsoleModes {
-	ICONSOLE_FULL,   ///< In-game console is closed.
+	ICONSOLE_FULL,   ///< In-game console is opened, whole screen.
 	ICONSOLE_OPENED, ///< In-game console is opened, upper 1/3 of the screen.
-	ICONSOLE_CLOSED, ///< In-game console is opened, whole screen.
+	ICONSOLE_CLOSED, ///< In-game console is closed.
 };
 
 /* Colours of the console messages. */
-- 
2.55.0

