From c06175cf27ac75c5bafa7baad4114bf628b6351f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Sun, 3 Oct 2021 10:49:03 +0200
Subject: [PATCH 0598/1053] Fix: Incorrect Romanian own name (#9598)

---
 src/lang/romanian.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lang/romanian.txt b/src/lang/romanian.txt
index ea10458d7f..dfb9af934c 100644
--- a/src/lang/romanian.txt
+++ b/src/lang/romanian.txt
@@ -1,5 +1,5 @@
 ##name Romanian
-##ownname Românӑ
+##ownname Română
 ##isocode ro_RO
 ##plural 14
 ##textdir ltr
-- 
2.55.0

