From fc7ed2591500ac87fdba768af79b8f83fcaea4b0 Mon Sep 17 00:00:00 2001
From: Charles Pigott <charlespigott@googlemail.com>
Date: Sun, 17 Oct 2021 18:44:39 +0100
Subject: [PATCH 0612/1053] Change: Ignore refresh_rate setting when vsync is
 enabled

---
 src/video/video_driver.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/video/video_driver.hpp b/src/video/video_driver.hpp
index b87dc70529..458c674024 100644
--- a/src/video/video_driver.hpp
+++ b/src/video/video_driver.hpp
@@ -316,6 +316,8 @@ protected:
 
 	std::chrono::steady_clock::duration GetDrawInterval()
 	{
+		/* If vsync, draw interval is decided by the display driver */
+		if (_video_vsync && _video_hw_accel) return std::chrono::microseconds(0);
 		return std::chrono::microseconds(1000000 / _settings_client.gui.refresh_rate);
 	}
 
-- 
2.55.0

