From f13052b742ec65fbd482a45888fa89aeb7dd3ef4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Fri, 22 Oct 2021 16:34:48 +0200
Subject: [PATCH 0621/1053] Fix #9643, 95386dc: Incorrect determination of
 screenshot format (#9644)

---
 src/screenshot.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/screenshot.cpp b/src/screenshot.cpp
index 244aede50f..f91648cf47 100644
--- a/src/screenshot.cpp
+++ b/src/screenshot.cpp
@@ -584,7 +584,7 @@ void InitializeScreenshotFormats()
 {
 	uint j = 0;
 	for (uint i = 0; i < lengthof(_screenshot_formats); i++) {
-		if (_screenshot_format_name.compare(_screenshot_formats[i].extension) != 0) {
+		if (_screenshot_format_name.compare(_screenshot_formats[i].extension) == 0) {
 			j = i;
 			break;
 		}
-- 
2.55.0

