From 5a472beaa1ec6337cc13fc4fc4d95d0ee2a7ba06 Mon Sep 17 00:00:00 2001
From: Dmitry Erlikh <derlih@gmail.com>
Date: Wed, 3 Nov 2021 22:10:21 +0100
Subject: [PATCH 0626/1053] Fix #9648: add missing commas in crash message

---
 src/saveload/afterload.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/saveload/afterload.cpp b/src/saveload/afterload.cpp
index 6a42ea81c6..d3194cd4e2 100644
--- a/src/saveload/afterload.cpp
+++ b/src/saveload/afterload.cpp
@@ -393,8 +393,8 @@ static void CDECL HandleSavegameLoadCrash(int signum)
 			"or older version.\n"
 			"It will load a NewGRF with the same GRF ID as the missing NewGRF.\n"
 			"This means that if the author makes incompatible NewGRFs with the\n"
-			"same GRF ID OpenTTD cannot magically do the right thing. In most\n"
-			"cases OpenTTD will load the savegame and not crash, but this is an\n"
+			"same GRF ID, OpenTTD cannot magically do the right thing. In most\n"
+			"cases, OpenTTD will load the savegame and not crash, but this is an\n"
 			"exception.\n"
 			"Please load the savegame with the appropriate NewGRFs installed.\n"
 			"The missing/compatible NewGRFs are:\n");
-- 
2.55.0

