From dde5c130ae828a9ae461c8a7f565db1dc602b51c Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Sun, 2 May 2021 23:36:55 +0300
Subject: [PATCH 0654/1053] Emscripten: small fix to build script

---
 os/emscripten/emscripten-build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os/emscripten/emscripten-build.sh b/os/emscripten/emscripten-build.sh
index c8f45aa1b2..db6d17a677 100755
--- a/os/emscripten/emscripten-build.sh
+++ b/os/emscripten/emscripten-build.sh
@@ -59,4 +59,4 @@ mkdir -p baseset
 [ -e Makefile ] || emcmake cmake .. -DHOST_BINARY_DIR=$(pwd)/build-host -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DOPTION_USE_ASSERTS=OFF || exit 1
 emmake make -j8 VERBOSE=1 || exit 1
 
-cp -f *.html *.js *.mem *.data *.wasm ../media/openttd.256.png ../os/emscripten/openttd.webapp /var/www/html/
+cp -f *.html *.js *.data *.wasm ../media/openttd.256.png ../os/emscripten/openttd.webapp /var/www/html/
-- 
2.55.0

