From 74066154d7495e238392d5970933b2c40f73e56f Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Mon, 3 May 2021 00:26:35 +0300
Subject: [PATCH 0658/1053] Emscripten: updated app manifest

---
 os/emscripten/openttd.webapp | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/os/emscripten/openttd.webapp b/os/emscripten/openttd.webapp
index 37a8be787a..7a5318912b 100644
--- a/os/emscripten/openttd.webapp
+++ b/os/emscripten/openttd.webapp
@@ -1,13 +1,16 @@
 {
-    "name":         "OpenTTD",
-    "short_name":   "OpenTTD",
-    "description":  "Open-source transport business simulation game",
-    "display":      "fullscreen",
-    "orientation":  "landscape",
+    "name":             "OpenTTD",
+    "short_name":       "OpenTTD",
+    "description":      "Open-source transport business simulation game",
+    "display":          "fullscreen",
+    "orientation":      "landscape",
     "icons": [
         {
-            "src": "openttd.256.png",
-            "sizes": "256x256"
+            "src":      "openttd.256.png",
+            "sizes":    "256x256",
+            "type":     "image/png"
         }
-    ]
+    ],
+    "start_url":        "/openttd.html",
+    "background_color": "green"
 }
-- 
2.55.0

