From f2812cd7e36e2e8b4711115421e5d8ebfe54c325 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 5 May 2021 00:50:59 +0300
Subject: [PATCH 0669/1053] Fixed saving config file

---
 os/emscripten/pre.js | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/os/emscripten/pre.js b/os/emscripten/pre.js
index c39220406f..23b33b950e 100644
--- a/os/emscripten/pre.js
+++ b/os/emscripten/pre.js
@@ -13,18 +13,20 @@ Module['websocket'] = { url: function(host, port, proto) {
 } };
 
 Module.preRun.push(function() {
-    personal_dir = '/home/web_user/.local/share/openttd';
-    content_download_dir = personal_dir + '/content_download'
+    const personal_dir = '/home/web_user/.local/share/openttd';
+    const content_download_dir = personal_dir + '/content_download'
 
     /* Because of the "-c" above, all user-data is stored in /user_data. */
-    FS.mkdir('/home/web_user/.local');
-    FS.mkdir('/home/web_user/.local/share');
-    FS.mkdir(personal_dir);
-    FS.mount(IDBFS, {}, personal_dir);
+    FS.mount(IDBFS, {}, '/home/web_user');
 
     Module.addRunDependency('syncfs');
     FS.syncfs(true, function (err) {
         /* FS.mkdir() tends to fail if parent folders do not exist. */
+        if (!FS.analyzePath(personal_dir).exists) {
+            FS.mkdir('/home/web_user/.local');
+            FS.mkdir('/home/web_user/.local/share');
+            FS.mkdir(personal_dir);
+        }
         if (!FS.analyzePath(content_download_dir).exists) {
             FS.mkdir(content_download_dir);
         }
-- 
2.55.0

