From 87dded477d4a360df4fa777a3e348cafa16af557 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 5 May 2021 01:45:58 +0300
Subject: [PATCH 0670/1053] Emscripten: Try smaller audio buffer

---
 src/sound/sdl2_s.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sound/sdl2_s.cpp b/src/sound/sdl2_s.cpp
index 7727f2f9a5..98e9d75a34 100644
--- a/src/sound/sdl2_s.cpp
+++ b/src/sound/sdl2_s.cpp
@@ -53,7 +53,7 @@ const char *SoundDriver_SDL::Start(const StringList &parm)
 	// Yes, allocate 1/3 of a second for an audiobuffer.
 	// Web browsers process sounds in the main thread together with video,
 	// and FPS of the webapp is terrible.
-	spec.samples = GetDriverParamInt(parm, "samples", 16384);
+	spec.samples = GetDriverParamInt(parm, "samples", 4096);
 #endif
 	spec.callback = fill_sound_buffer;
 	SDL_AudioDeviceID dev = SDL_OpenAudioDevice(nullptr, 0, &spec, &spec_actual, SDL_AUDIO_ALLOW_FREQUENCY_CHANGE);
-- 
2.55.0

