From 82b8772f8fa09010f4b3fcc33ac2045607353def Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Wed, 5 May 2021 23:49:21 +0300
Subject: [PATCH 0680/1053] Emscripten: enable debug logs

---
 os/emscripten/pre.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os/emscripten/pre.js b/os/emscripten/pre.js
index 23b33b950e..113a3acc10 100644
--- a/os/emscripten/pre.js
+++ b/os/emscripten/pre.js
@@ -1,4 +1,4 @@
-// Module.arguments.push('-mnull', '-snull', '-vsdl:relative_mode');
+Module.arguments.push('-d1');
 Module['websocket'] = { url: function(host, port, proto) {
     /* openttd.org hosts a WebSocket proxy for the content service. */
     if (host == "content.openttd.org" && port == 3978 && proto == "tcp") {
-- 
2.55.0

