From e84772f882c2a57027087167c32675ccbd097589 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Fri, 7 May 2021 21:41:59 +0300
Subject: [PATCH 0689/1053] Updated revision file

---
 .ottdrev | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.ottdrev b/.ottdrev
index 20c6f71741..5ca22e1b18 100644
--- a/.ottdrev
+++ b/.ottdrev
@@ -1 +1 @@
-1.11.1	20210418	0	0be22efffc3c14db08baf5e58c448b5d074f4427	1	1	2021
+1.11.2	20210503	0	672f285218c6817784d86f737987b75db4bc78fc	1	1	2021
-- 
2.55.0

