From 7a488b09ea63aa414f97f21f4ed0608a114dd40c Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Sat, 8 May 2021 00:03:20 +0300
Subject: [PATCH 0693/1053] Fixed mutex deadlock in libtimidity shutdown
 function

---
 src/music/libtimidity.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/music/libtimidity.cpp b/src/music/libtimidity.cpp
index 2c2795a188..3c2c419e09 100644
--- a/src/music/libtimidity.cpp
+++ b/src/music/libtimidity.cpp
@@ -82,11 +82,12 @@ const char *MusicDriver_LibTimidity::Start(const StringList &param)
 
 void MusicDriver_LibTimidity::Stop()
 {
+	this->StopSong();
+
 	MxSetMusicSource(nullptr);
 
 	std::lock_guard<std::mutex> lock{ _midi.synth_mutex };
 
-	if (_midi.status == MIDI_PLAYING) this->StopSong();
 	mid_exit();
 }
 
-- 
2.55.0

