From 6c08881b4db4f42163b782855ac49517aca2da62 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Sat, 15 May 2021 23:44:25 +0300
Subject: [PATCH 0704/1053] Removed duplicate setting

---
 src/table/settings.ini | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/table/settings.ini b/src/table/settings.ini
index af8e6722a1..98d045efda 100644
--- a/src/table/settings.ini
+++ b/src/table/settings.ini
@@ -2688,15 +2688,6 @@ strhelp  = STR_CONFIG_SETTING_AUTOSAVE_HELPTEXT
 strval   = STR_GAME_OPTIONS_AUTOSAVE_DROPDOWN_OFF
 cat      = SC_BASIC
 
-[SDTC_OMANY]
-var      = gui.save_to_network
-type     = SLE_UINT8
-flags    = SLF_NOT_IN_SAVE | SLF_NO_NETWORK_SYNC
-def      = 2
-max      = 2
-full     = _save_to_network
-cat      = SC_BASIC
-
 [SDTC_BOOL]
 var      = gui.threaded_saves
 flags    = SLF_NOT_IN_SAVE | SLF_NO_NETWORK_SYNC
-- 
2.55.0

