From 3807e395474527a6f5c880bd779863ab90205ce1 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 24 Jun 2021 00:38:31 +0300
Subject: [PATCH 0718/1053] Fixed console obscured by vertical toolbar

---
 src/console_gui.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/console_gui.cpp b/src/console_gui.cpp
index 17458a1c9e..0dab2279df 100644
--- a/src/console_gui.cpp
+++ b/src/console_gui.cpp
@@ -187,7 +187,8 @@ struct IConsoleWindow : Window
 
 		this->InitNested(0);
 		this->truncate_timer.SetInterval(3000);
-		ResizeWindow(this, _screen.width, _screen.height / 3);
+		ResizeWindow(this, _screen.width - GetMinButtonSize() * 2, _screen.height / 3);
+		this->left = GetMinButtonSize();
 	}
 
 	~IConsoleWindow()
@@ -436,11 +437,11 @@ void IConsoleResize(Window *w)
 	switch (_iconsole_mode) {
 		case ICONSOLE_OPENED:
 			w->height = _screen.height / 3;
-			w->width = _screen.width;
+			w->width = _screen.width - GetMinButtonSize() * 2;
 			break;
 		case ICONSOLE_FULL:
 			w->height = _screen.height - ICON_BOTTOM_BORDERWIDTH;
-			w->width = _screen.width;
+			w->width = _screen.width - GetMinButtonSize() * 2;
 			break;
 		default: return;
 	}
-- 
2.55.0

