From 339b89f8314e8ce29d10e543a1ea447ecf212d0d Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 8 Jul 2021 00:22:25 +0300
Subject: [PATCH 0724/1053] Emscripten: Fixed compilation

---
 src/video/sdl2_v.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/video/sdl2_v.cpp b/src/video/sdl2_v.cpp
index 5941c1979d..773934db9f 100644
--- a/src/video/sdl2_v.cpp
+++ b/src/video/sdl2_v.cpp
@@ -624,6 +624,7 @@ void VideoDriver_SDL_Base::InputLoop()
 {
 	const Uint8 *keys = SDL_GetKeyboardState(NULL);
 
+#ifndef __EMSCRIPTEN__
 #if defined(_DEBUG)
 	this->fast_forward_key_pressed = _shift_pressed;
 #else
@@ -631,6 +632,7 @@ void VideoDriver_SDL_Base::InputLoop()
 	 * to switch to another application. */
 	this->fast_forward_key_pressed = keys[SDL_SCANCODE_TAB] && (mod & KMOD_ALT) == 0;
 #endif /* defined(_DEBUG) */
+#endif // __EMSCRIPTEN__
 
 	/* Determine which directional keys are down. */
 	_dirkeys =
-- 
2.55.0

