From 47b01a0528c726d5be8d63a7ed652d5ed9938d47 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Sun, 22 Aug 2021 00:28:05 +0300
Subject: [PATCH 0726/1053] Fixed compilation

---
 src/video/sdl_v.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/video/sdl_v.cpp b/src/video/sdl_v.cpp
index 73e3db5670..ce6289f92a 100644
--- a/src/video/sdl_v.cpp
+++ b/src/video/sdl_v.cpp
@@ -695,6 +695,7 @@ void VideoDriver_SDL::Stop()
 
 void VideoDriver_SDL::InputLoop()
 {
+	uint32 mod = SDL_GetModState();
 	int numkeys;
 	Uint8 *keys = SDL_GetKeyState(&numkeys);
 
-- 
2.55.0

