From 00787f52b2075869dc5049851ad8595a04c3eca3 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Thu, 2 Dec 2021 02:52:20 +0200
Subject: [PATCH 0735/1053] Fixed a crash when closing error window

---
 src/error_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/error_gui.cpp b/src/error_gui.cpp
index 6ebbfd7e90..6f32f0676f 100644
--- a/src/error_gui.cpp
+++ b/src/error_gui.cpp
@@ -335,7 +335,7 @@ public:
 	{
 		switch (widget) {
 			case WID_EM_CLOSE:
-				delete this;
+				this->Close();
 				break;
 
 			default:
-- 
2.55.0

