From aee14799a62519cd133e52a60cb1fdbfc6ee6227 Mon Sep 17 00:00:00 2001
From: Sergii Pylypenko <x.pelya.x@gmail.com>
Date: Mon, 20 Dec 2021 01:47:46 +0200
Subject: [PATCH 0760/1053] Android: fixed two finger scrolling bug

---
 src/video/sdl_v.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/video/sdl_v.cpp b/src/video/sdl_v.cpp
index 11874e9ee1..4a3926e72b 100644
--- a/src/video/sdl_v.cpp
+++ b/src/video/sdl_v.cpp
@@ -534,8 +534,8 @@ bool VideoDriver_SDL::PollEvent()
 				case SDL_BUTTON_RIGHT:
 					_right_button_down = true;
 					_right_button_clicked = true;
-					_right_button_down_pos.x = ev.motion.x;
-					_right_button_down_pos.y = ev.motion.y;
+					_right_button_down_pos.x = ev.button.x;
+					_right_button_down_pos.y = ev.button.y;
 					break;
 
 #ifdef __ANDROID__
-- 
2.55.0

