From 280f54c0f735d53039d250bf3725a3e385374a3f Mon Sep 17 00:00:00 2001
From: Owen Rudge <owen@owenrudge.net>
Date: Thu, 11 Nov 2021 19:06:31 +0000
Subject: [PATCH 0768/1053] Add: Specify minimum version for macOS (#9689)

---
 os/macosx/Info.plist.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os/macosx/Info.plist.in b/os/macosx/Info.plist.in
index 359d28de8d..0148ab289a 100644
--- a/os/macosx/Info.plist.in
+++ b/os/macosx/Info.plist.in
@@ -31,5 +31,7 @@
         <string>NSApplication</string>
         <key>NSHighResolutionCapable</key>
         <string>True</string>
+        <key>LSMinimumSystemVersion</key>
+        <string>10.14.0</string>
 </dict>
 </plist>
-- 
2.55.0

