From b185c80cb80cc64aff6ede0dafa09f80239b1238 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Tue, 28 Dec 2021 21:08:09 +0100
Subject: [PATCH 0783/1053] Fix: [MinGW] Disable ASLR for x64 Debug builds
 (#9773)

---
 cmake/CompileFlags.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cmake/CompileFlags.cmake b/cmake/CompileFlags.cmake
index 659a9ca724..ad4a46aca7 100644
--- a/cmake/CompileFlags.cmake
+++ b/cmake/CompileFlags.cmake
@@ -43,6 +43,11 @@ macro(compile_flags)
         add_link_options(
             "$<$<NOT:$<CONFIG:Debug>>:-fstack-protector>" # Prevent undefined references when _FORTIFY_SOURCE > 0
         )
+        if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+            add_link_options(
+                "$<$<CONFIG:Debug>:-Wl,--disable-dynamicbase,--disable-high-entropy-va,--default-image-base-low>" # ASLR somehow breaks linking for x64 Debug builds
+            )
+        endif()
     endif()
 
     # Prepare a generator that checks if we are not a debug, and don't have asserts
-- 
2.55.0

