From 46b8382c64dd94a606824c31c05ddae000f876c5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Tue, 4 Jan 2022 22:12:05 +0100
Subject: [PATCH 0803/1053] Fix #9766: Don't write uninitialised data in config
 file (#9767)

---
 src/settings.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/settings.cpp b/src/settings.cpp
index 52f30be082..2dc12fb144 100644
--- a/src/settings.cpp
+++ b/src/settings.cpp
@@ -356,6 +356,10 @@ void OneOfManySettingDesc::FormatValue(char *buf, const char *last, const void *
 void ManyOfManySettingDesc::FormatValue(char *buf, const char *last, const void *object) const
 {
 	uint bitmask = (uint)this->Read(object);
+	if (bitmask == 0) {
+		buf[0] = '\0';
+		return;
+	}
 	bool first = true;
 	for (uint id : SetBitIterator(bitmask)) {
 		if (!first) buf = strecpy(buf, "|", last);
-- 
2.55.0

