From 3a7d2f32a4112d0ce2a22ad8dc3bab9da587e62a Mon Sep 17 00:00:00 2001
From: ALEX11BR <49609151+ALEX11BR@users.noreply.github.com>
Date: Fri, 18 Feb 2022 16:29:39 +0200
Subject: [PATCH 0816/1053] Fix: correct some mistaken Romanian real towns

---
 src/table/townname.h | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/table/townname.h b/src/table/townname.h
index 85ac330609..b3f197a046 100644
--- a/src/table/townname.h
+++ b/src/table/townname.h
@@ -1943,7 +1943,7 @@ static const char * const _name_romanian_real[] = {
 	u8"Bac\u0103u",
 	"Baia Mare",
 	u8"B\u0103ile Herculane",
-	u8"B\u0103ilesti",
+	u8"B\u0103ile\u0219ti",
 	u8"B\u00e2rlad",
 	"Bicaz",
 	u8"Bistri\u021ba",
@@ -1951,11 +1951,11 @@ static const char * const _name_romanian_real[] = {
 	"Borsec",
 	u8"Boto\u0219ani",
 	u8"Br\u0103ila",
-	u8"Bra\u021bov",
+	u8"Bra\u0219ov",
 	u8"Bucure\u0219ti",
 	"Buftea",
 	u8"Buz\u0103u",
-	u8"C\u0103l\u0103rasi",
+	u8"C\u0103l\u0103ra\u0219i",
 	u8"Caransebe\u0219",
 	u8"Cernavod\u0103",
 	"Cluj-Napoca",
@@ -1966,12 +1966,12 @@ static const char * const _name_romanian_real[] = {
 	"Deva",
 	"Dorohoi",
 	"Drobeta-Turnu Severin",
-	u8"Dr\u0103g\u0103sani",
-	u8"F\u0103g\u0103ras",
+	u8"Dr\u0103g\u0103\u0219ani",
+	u8"F\u0103g\u0103ra\u0219",
 	u8"F\u0103lticeni",
 	u8"Fete\u0219ti",
 	u8"Foc\u0219ani",
-	u8"Gala\u0219i",
+	u8"Gala\u021bi",
 	"Gheorgheni",
 	"Giurgiu",
 	u8"H\u00e2r\u0219ova",
@@ -2015,10 +2015,10 @@ static const char * const _name_romanian_real[] = {
 	"Sovata",
 	"Suceava",
 	"Sulina",
-	u8"T\u0103nd\u0103rei",
-	u8"T\u00e2rgoviste",
+	u8"\u021a\u0103nd\u0103rei",
+	u8"T\u00e2rgovi\u0219te",
 	u8"T\u00e2rgu Jiu",
-	u8"T\u00e2rgu Mures",
+	u8"T\u00e2rgu Mure\u0219",
 	"Tecuci",
 	u8"Timi\u0219oara",
 	"Tulcea",
-- 
2.55.0

