From a637080136e21fab47061e82f84310d00f78dcf4 Mon Sep 17 00:00:00 2001
From: Miguel Horta <miguel.horta@petrotec.com>
Date: Tue, 12 Apr 2022 13:56:24 +0100
Subject: [PATCH 0823/1053] Bump .ottdrev to 12.2

---
 .ottdrev | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.ottdrev b/.ottdrev
index de6bbcb484..57e51a44df 100644
--- a/.ottdrev
+++ b/.ottdrev
@@ -1 +1 @@
-12.1	20211108	0	17dfc6e28aac5ad2290e849b03893a416be97b03	1	1	2021
+12.2	20220412	0	1893dba8a172636791dc79137a3331bd41cc50bc	1	1	2022
-- 
2.55.0

