From b211a88b71aba77164799d2fef64cde5913dee45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Mon, 2 Jan 2023 02:13:16 +0100
Subject: [PATCH 0830/1053] Fix #10304, fe30f66: [Scripts] Don't start GS in
 intro (#10305)

---
 src/game/game_core.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/game/game_core.cpp b/src/game/game_core.cpp
index c5ba56e460..caa57e1322 100644
--- a/src/game/game_core.cpp
+++ b/src/game/game_core.cpp
@@ -73,6 +73,9 @@
 {
 	if (Game::instance != nullptr) return;
 
+	/* Don't start GameScripts in intro */
+	if (_game_mode == GM_MENU) return;
+
 	/* Clients shouldn't start GameScripts */
 	if (_networking && !_network_server) return;
 
-- 
2.55.0

