From f37b6cb17ff93e2b500de0d60ef26d2629f3d847 Mon Sep 17 00:00:00 2001
From: Rubidium <rubidium@openttd.org>
Date: Sun, 15 Jan 2023 10:53:32 +0100
Subject: [PATCH 0843/1053] Fix #9865: removing files with the console always
 failed

---
 src/console_cmds.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/console_cmds.cpp b/src/console_cmds.cpp
index 33b47d9600..05d361d778 100644
--- a/src/console_cmds.cpp
+++ b/src/console_cmds.cpp
@@ -454,8 +454,8 @@ DEF_CONSOLE_CMD(ConRemove)
 	_console_file_list.ValidateFileList();
 	const FiosItem *item = _console_file_list.FindItem(file);
 	if (item != nullptr) {
-		if (!FiosDelete(item->name)) {
-			IConsolePrint(CC_ERROR, "Failed to delete '{}'.", file);
+		if (unlink(item->name) != 0) {
+			IConsolePrint(CC_ERROR, "Failed to delete '{}'.", item->name);
 		}
 	} else {
 		IConsolePrint(CC_ERROR, "'{}' could not be found.", file);
-- 
2.55.0

