From 017d569fe8333b9f0aacc6410ed86ac4426d3f52 Mon Sep 17 00:00:00 2001
From: chucky-n-ice <chucky_n-ice@outlook.com>
Date: Sun, 3 Dec 2023 17:57:21 +0100
Subject: [PATCH 0878/1053] Fixme: Error while try to compile!

---
 src/tutorial_gui.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/tutorial_gui.cpp b/src/tutorial_gui.cpp
index 3f6aa7e4b4..a822ecb04b 100644
--- a/src/tutorial_gui.cpp
+++ b/src/tutorial_gui.cpp
@@ -92,14 +92,14 @@ void OpenExternTutorialVideo(VideoLink_t *tutorial)
 #ifdef __ANDROID__
 	SDL_ANDROID_OpenExternalWebBrowser(link);
 #else
-	char cmd[PATH_MAX] =
+	//char cmd[PATH_MAX] =
 #ifdef WIN32
 		"start ";
 #else
 		"xdg-open ";
 #endif
-	strcat(cmd, link);
-	system(cmd);
+	//strcat(cmd, link);
+	//system(cmd);
 #endif
 }
 
-- 
2.55.0

