From 0e2f7614eae2da56d09256a1a630603bb614ba1b Mon Sep 17 00:00:00 2001
From: chucky-n-ice <chucky_n-ice@outlook.com>
Date: Sun, 3 Dec 2023 18:21:28 +0100
Subject: [PATCH 0879/1053] update version info

---
 .ottdrev | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.ottdrev b/.ottdrev
index 57e51a44df..88a3eab94e 100644
--- a/.ottdrev
+++ b/.ottdrev
@@ -1 +1 @@
-12.2	20220412	0	1893dba8a172636791dc79137a3331bd41cc50bc	1	1	2022
+13.0	20231203	0	2ebc601d97aa431e5d8aebfe4a7bbfdabf29dc75	1	1	2023
-- 
2.55.0

