From 5a018571be17cb8423147b79126738c3d4e1e7ae Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Thu, 4 Apr 2024 15:48:41 +0100
Subject: [PATCH 0883/1053] fix: Add vertical toolbars to always shown windows

---
 src/toolbar_gui.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/toolbar_gui.cpp b/src/toolbar_gui.cpp
index 5fc3740c46..17c8ce1a50 100644
--- a/src/toolbar_gui.cpp
+++ b/src/toolbar_gui.cpp
@@ -2947,7 +2947,7 @@ static const NWidgetPart _nested_toolbar_vertical_left_widgets[] = {
 static WindowDesc _toolb_vertical_left_desc(__FILE__, __LINE__,
 	WDP_MANUAL, nullptr, 22, 480,
 	WC_MAIN_TOOLBAR, WC_NONE,
-	WDF_NO_FOCUS,
+	WDF_NO_FOCUS | WDF_NO_CLOSE,
 	std::begin(_nested_toolbar_vertical_left_widgets), std::end(_nested_toolbar_vertical_left_widgets),
 	&MainToolbarWindow::hotkeys
 );
@@ -2973,7 +2973,7 @@ static const NWidgetPart _nested_toolbar_vertical_right_widgets[] = {
 static WindowDesc _toolb_vertical_right_desc(__FILE__, __LINE__,
 	WDP_MANUAL, nullptr, 22, 480,
 	WC_MAIN_TOOLBAR_RIGHT, WC_NONE,
-	WDF_NO_FOCUS,
+	WDF_NO_FOCUS | WDF_NO_CLOSE,
 	std::begin(_nested_toolbar_vertical_right_widgets), std::end(_nested_toolbar_vertical_right_widgets),
 	&MainToolbarWindow::hotkeys
 );
-- 
2.55.0

