From 52a3a63059c860759edc3247614e87741a5dc68f Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Thu, 2 Mar 2023 16:04:47 +0000
Subject: [PATCH 0884/1053] fix: use new console interface

---
 src/console_gui.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/console_gui.cpp b/src/console_gui.cpp
index 1567cb4352..edd3e9bf8c 100644
--- a/src/console_gui.cpp
+++ b/src/console_gui.cpp
@@ -397,9 +397,11 @@ void IConsoleSwitch()
 #ifdef __ANDROID__
 			{
 				char buf[1024] = "";
-				for (const IConsoleLine *print = IConsoleLine::Get(0); print != NULL; print = print->previous) {
-					if (print->buffer && print->buffer[0]) {
-						strecat(buf, print->buffer, lastof(buf));
+
+				//for (const IConsoleLine *print = IConsoleLine::Get(0); print != NULL; print = print->previous) {
+				for (IConsoleLine &line : _iconsole_buffer) {
+					if (!line.buffer.empty()) {
+						strecat(buf, line.buffer.c_str(), lastof(buf));
 						strecat(buf, "\n", lastof(buf));
 					}
 				}
-- 
2.55.0

