From 120851eaf91649e87b9c4c083e3008af28e7e24b Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Sun, 9 Apr 2023 23:17:00 +0100
Subject: [PATCH 0889/1053] fix: dock cmd was not called

---
 src/dock_gui.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/dock_gui.cpp b/src/dock_gui.cpp
index 39a386c85e..71a2990d7a 100644
--- a/src/dock_gui.cpp
+++ b/src/dock_gui.cpp
@@ -233,6 +233,7 @@ struct BuildDocksToolbarWindow : Window {
 	{
 		switch (last_clicked_widget) {
 			case WID_DT_BUILD_AQUEDUCT:
+			case WID_DT_LOCK:
 			case WID_DT_STATION:
 				this->OnPlacePresize(pt, TileVirtXY(pt.x, pt.y));
 				break;
@@ -268,6 +269,7 @@ struct BuildDocksToolbarWindow : Window {
 						    return Command<CMD_BUILD_DOCK>::Post(STR_ERROR_CAN_T_BUILD_DOCK_HERE, CcBuildDocks, end_tile, to_join, adjacent);
 					    }
 				    };
+					ShowSelectStationIfNeeded(TileArea(start_tile, end_tile), proc);
 					VpStartPreSizing();
 					break;
 				}
-- 
2.55.0

