From c34122aeff9ae468d0efe6fef6c36bc1a48ff994 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Sun, 9 Apr 2023 23:37:26 +0100
Subject: [PATCH 0890/1053] fix: typo

---
 src/dock_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dock_gui.cpp b/src/dock_gui.cpp
index 71a2990d7a..421328ed61 100644
--- a/src/dock_gui.cpp
+++ b/src/dock_gui.cpp
@@ -269,7 +269,7 @@ struct BuildDocksToolbarWindow : Window {
 						    return Command<CMD_BUILD_DOCK>::Post(STR_ERROR_CAN_T_BUILD_DOCK_HERE, CcBuildDocks, end_tile, to_join, adjacent);
 					    }
 				    };
-					ShowSelectStationIfNeeded(TileArea(start_tile, end_tile), proc);
+					ShowSelectStationIfNeeded(TileArea(start_tile, tile_to), proc);
 					VpStartPreSizing();
 					break;
 				}
-- 
2.55.0

