From ee1fcd4b1766c7f2823c75ca624fc25a6d63131f Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Mon, 10 Apr 2023 00:11:52 +0100
Subject: [PATCH 0891/1053] fix: swap dock tiles

---
 src/dock_gui.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/dock_gui.cpp b/src/dock_gui.cpp
index 421328ed61..45d7d6ad65 100644
--- a/src/dock_gui.cpp
+++ b/src/dock_gui.cpp
@@ -258,15 +258,15 @@ struct BuildDocksToolbarWindow : Window {
 					break;
 				case DDSP_BUILD_STATION: {
 				    /* Determine the watery part of the dock. */
-				    DiagDirection dir = GetInclinedSlopeDirection(GetTileSlope(end_tile));
-				    TileIndex tile_to = (dir != INVALID_DIAGDIR ? TileAddByDiagDir(end_tile, ReverseDiagDir(dir)) : end_tile);
+				    DiagDirection dir = GetInclinedSlopeDirection(GetTileSlope(start_tile));
+				    TileIndex tile_to = (dir != INVALID_DIAGDIR ? TileAddByDiagDir(start_tile, ReverseDiagDir(dir)) : start_tile);
 
 				    bool adjacent = _ctrl_pressed;
 				    auto proc = [=](bool test, StationID to_join) -> bool {
 					    if (test) {
-						    return Command<CMD_BUILD_DOCK>::Do(CommandFlagsToDCFlags(GetCommandFlags<CMD_BUILD_DOCK>()), end_tile, INVALID_STATION, adjacent).Succeeded();
+						    return Command<CMD_BUILD_DOCK>::Do(CommandFlagsToDCFlags(GetCommandFlags<CMD_BUILD_DOCK>()), start_tile, INVALID_STATION, adjacent).Succeeded();
 					    } else {
-						    return Command<CMD_BUILD_DOCK>::Post(STR_ERROR_CAN_T_BUILD_DOCK_HERE, CcBuildDocks, end_tile, to_join, adjacent);
+						    return Command<CMD_BUILD_DOCK>::Post(STR_ERROR_CAN_T_BUILD_DOCK_HERE, CcBuildDocks, start_tile, to_join, adjacent);
 					    }
 				    };
 					ShowSelectStationIfNeeded(TileArea(start_tile, tile_to), proc);
-- 
2.55.0

