From 5d36d9ef592ee7bbdf438eee2bd5321e8a1bf804 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Sat, 6 Apr 2024 14:16:03 +0100
Subject: [PATCH 0895/1053] fix: Restore Android entrypoint

---
 src/os/unix/unix_main.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/os/unix/unix_main.cpp b/src/os/unix/unix_main.cpp
index 41ad22ced5..01c6f43853 100644
--- a/src/os/unix/unix_main.cpp
+++ b/src/os/unix/unix_main.cpp
@@ -18,6 +18,11 @@
 
 #include "../../safeguards.h"
 
+#ifdef __ANDROID__
+#define main SDL_main
+extern "C" int CDECL main(int, char *[]);
+#endif
+
 int CDECL main(int argc, char *argv[])
 {
 	/* Make sure our arguments contain only valid UTF-8 characters. */
-- 
2.55.0

