From 9e1f6606471419d2f57d4e5ec4b039fcc5c0ba40 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Sun, 7 Apr 2024 13:45:03 +0100
Subject: [PATCH 0900/1053] fix: OSOpenBrowser now takes a string

---
 src/os/unix/unix.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/os/unix/unix.cpp b/src/os/unix/unix.cpp
index 7426d7f987..4e6ff8cee8 100644
--- a/src/os/unix/unix.cpp
+++ b/src/os/unix/unix.cpp
@@ -257,7 +257,7 @@ void OSOpenBrowser(const std::string &url)
 	args[4] = "--user";
 	args[5] = "0";
 	args[6] = "-d";
-	args[7] = url;
+	args[7] = url.c_str();
 	args[8] = NULL;
 #else
 	const char *args[3];
-- 
2.55.0

