From 9e89acb6327275f1203162af7bcd967fbaf443e7 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Mon, 8 Apr 2024 10:17:44 +0100
Subject: [PATCH 0901/1053] fix: ShowDropDownList*() signature changed, update
 usages

---
 src/toolbar_gui.cpp | 28 ++++++++++++++++++----------
 1 file changed, 18 insertions(+), 10 deletions(-)

diff --git a/src/toolbar_gui.cpp b/src/toolbar_gui.cpp
index a1fd5312f7..c302ad5b68 100644
--- a/src/toolbar_gui.cpp
+++ b/src/toolbar_gui.cpp
@@ -118,7 +118,7 @@ public:
 static void PopupMainToolbarMenu(Window *w, WidgetID widget, DropDownList &&list, int def)
 {
 	if (!_settings_client.gui.vertical_toolbar) {
-	ShowDropDownList(w, std::move(list), def, widget, 0, true);
+	ShowDropDownList(w, std::move(list), def, widget, 0, list.size() <= 1);
 	} else {
 		Rect wi_rect;
 		NWidgetCore *nwi = w->GetWidget<NWidgetCore>(widget);
@@ -126,7 +126,7 @@ static void PopupMainToolbarMenu(Window *w, WidgetID widget, DropDownList &&list
 		wi_rect.right  = nwi->pos_x + nwi->current_x;
 		wi_rect.top    = nwi->pos_y;
 		wi_rect.bottom = nwi->pos_y + nwi->current_y;
-		ShowDropDownListAt(w, std::move(list), def, widget, wi_rect, nwi->colour, true, list.size() <= 1);
+		ShowDropDownListAt(w, std::move(list), def, widget, wi_rect, nwi->colour, list.size() <= 1);
 	}
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 }
@@ -295,7 +295,7 @@ static CallBackFunction ToolbarOptionsClick(Window *w)
 	list.push_back(MakeDropDownListCheckedItem(IsTransparencySet(TO_HOUSES), STR_SETTINGS_MENU_TRANSPARENT_BUILDINGS, OME_TRANSPARENTBUILDINGS));
 	list.push_back(MakeDropDownListCheckedItem(IsTransparencySet(TO_SIGNS), STR_SETTINGS_MENU_TRANSPARENT_SIGNS, OME_SHOW_STATIONSIGNS));
 
-	ShowDropDownList(w, std::move(list), 0, WID_TN_SETTINGS, 140, true);
+	ShowDropDownList(w, std::move(list), 0, WID_TN_SETTINGS, 140, false);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 	return CBF_NONE;
 }
@@ -680,7 +680,7 @@ static CallBackFunction ToolbarGraphsClick(Window *w)
 
 	if (_toolbar_mode != TB_NORMAL) AddDropDownLeagueTableOptions(list);
 
-	ShowDropDownList(w, std::move(list), GRMN_OPERATING_PROFIT_GRAPH, WID_TN_GRAPHS, 140, true, false);
+	ShowDropDownList(w, std::move(list), GRMN_OPERATING_PROFIT_GRAPH, WID_TN_GRAPHS, 140, false);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 
 	return CBF_NONE;
@@ -693,7 +693,7 @@ static CallBackFunction ToolbarLeagueClick(Window *w)
 	AddDropDownLeagueTableOptions(list);
 
 	int selected = list[0]->result;
-	ShowDropDownList(w, std::move(list), selected, WID_TN_LEAGUE, 140, true, false);
+	ShowDropDownList(w, std::move(list), selected, WID_TN_LEAGUE, 140, false);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 
 	return CBF_NONE;
@@ -897,7 +897,7 @@ static CallBackFunction ToolbarBuildRailClick(Window *w)
 		list.emplace_back(new DropDownListIconItem(SPR_IMG_BUILDWATER, PAL_NONE, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, MENU_IDX_WATER, false));
 		list.emplace_back(new DropDownListIconItem(SPR_IMG_BUILDAIR, PAL_NONE, STR_AIRCRAFT_MENU_AIRPORT_CONSTRUCTION, MENU_IDX_AIR, false));
 	}
-	ShowDropDownList(w, std::move(list), _last_built_railtype, WID_TN_RAILS, 140, true);
+	ShowDropDownList(w, std::move(list), _last_built_railtype, WID_TN_RAILS, false);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 	return CBF_NONE;
 }
@@ -936,7 +936,9 @@ static CallBackFunction MenuClickBuildRail(int index)
 
 static CallBackFunction ToolbarBuildRoadClick(Window *w)
 {
-	ShowDropDownList(w, GetRoadTypeDropDownList(RTTB_ROAD), _last_built_roadtype, WID_TN_ROADS, 140, true);
+	auto list = GetScenRoadTypeDropDownList(RTTB_ROAD);
+	auto has_choices = list.size() <= 1;
+	ShowDropDownList(w, std::move(list), _last_built_roadtype, WID_TN_ROADS, 140, has_choices);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 	return CBF_NONE;
 }
@@ -958,7 +960,9 @@ static CallBackFunction MenuClickBuildRoad(int index)
 
 static CallBackFunction ToolbarBuildTramClick(Window *w)
 {
-	ShowDropDownList(w, GetRoadTypeDropDownList(RTTB_TRAM), _last_built_tramtype, WID_TN_TRAMS, 140, true);
+	auto list = GetScenRoadTypeDropDownList(RTTB_TRAM);
+	auto has_choices = list.size() <= 1;
+	ShowDropDownList(w, std::move(list), _last_built_tramtype, WID_TN_TRAMS, 140, has_choices);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 	return CBF_NONE;
 }
@@ -1329,7 +1333,9 @@ static CallBackFunction ToolbarScenGenIndustry(Window *w)
 
 static CallBackFunction ToolbarScenBuildRoadClick(Window *w)
 {
-	ShowDropDownList(w, GetScenRoadTypeDropDownList(RTTB_ROAD), _last_built_roadtype, WID_TE_ROADS, 140, true);
+	auto list = GetScenRoadTypeDropDownList(RTTB_ROAD);
+	auto has_choices = list.size() <= 1;
+	ShowDropDownList(w, std::move(list), _last_built_roadtype, WID_TE_ROADS, 140, has_choices);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 	return CBF_NONE;
 }
@@ -1349,7 +1355,9 @@ static CallBackFunction ToolbarScenBuildRoad(int index)
 
 static CallBackFunction ToolbarScenBuildTramClick(Window *w)
 {
-	ShowDropDownList(w, GetScenRoadTypeDropDownList(RTTB_TRAM), _last_built_tramtype, WID_TE_TRAMS, 140, true);
+	auto list = GetScenRoadTypeDropDownList(RTTB_TRAM);
+	auto has_choices = list.size() <= 1;
+	ShowDropDownList(w, std::move(list), _last_built_tramtype, WID_TE_TRAMS, 140, has_choices);
 	if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
 	return CBF_NONE;
 }
-- 
2.55.0

