From 9cba431edd1840f706c1889a74ab3c7e2ed8679e Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Wed, 5 Jun 2024 21:45:55 +0100
Subject: [PATCH 0920/1053] change: Revert viewport sizing logic

---
 src/widget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widget.cpp b/src/widget.cpp
index 1c07c28b58..d1604ef76b 100644
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -1034,7 +1034,7 @@ void NWidgetResizeBase::SetMinimalSizeForSizingType()
 			min_size = GetMinButtonSize();
 			break;
 		case NWST_VIEWPORT:
-			min_size = (5 * _gui_scale) / 2;
+			min_size = GetMinButtonSize() * 3;
 			break;
 		default: NOT_REACHED();
 	}
-- 
2.55.0

