From 71cffecb1337dc005b9ad98aceb39464d5a4b711 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Wed, 5 Jun 2024 23:22:38 +0100
Subject: [PATCH 0922/1053] fix: syntax error

---
 src/console_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/console_gui.cpp b/src/console_gui.cpp
index 1d5ae4623d..409afa9801 100644
--- a/src/console_gui.cpp
+++ b/src/console_gui.cpp
@@ -446,7 +446,7 @@ void IConsoleSwitch()
 #ifdef __ANDROID__
 			{
 				char buf[1024] = "";
-				char *pos = &buf;
+				char *pos = &buf[0];
 
 				for (IConsoleLine &line : _iconsole_buffer) {
 					if (!line.buffer.empty()) {
-- 
2.55.0

