From 394e7e4a58682b62902e0339ff58e5c4c9b74820 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Thu, 1 May 2025 00:24:31 +0100
Subject: [PATCH 0927/1053] fix crash when debug was enabled in cli

---
 src/debug.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/debug.cpp b/src/debug.cpp
index 23584ffa64..df926e8186 100644
--- a/src/debug.cpp
+++ b/src/debug.cpp
@@ -115,8 +115,8 @@ void DumpDebugFacilityNames(std::back_insert_iterator<std::string> &output_itera
 void DebugPrint(const char *category, int level, const std::string &message)
 {
 #ifdef __ANDROID__
-	__android_log_print(ANDROID_LOG_INFO, "OpenTTD", "[%s] %s", level, message.c_str());
-#endif
+	__android_log_print(ANDROID_LOG_INFO, "OpenTTD", "[%s:%d] %s", category, level, message.c_str());
+#else
 	if (strcmp(category, "desync") == 0) {
 		static FILE *f = FioFOpenFile("commands-out.log", "wb", AUTOSAVE_DIR);
 		if (f == nullptr) return;
@@ -140,6 +140,7 @@ void DebugPrint(const char *category, int level, const std::string &message)
 			_debug_remote_console_queue.push_back({ category, message });
 		}
 	}
+#endif
 }
 
 /**
-- 
2.55.0

