From dd835262edde42d549f3b841c63c21690d5ac9bf Mon Sep 17 00:00:00 2001
From: Rubidium <rubidium@openttd.org>
Date: Sat, 13 Dec 2025 23:07:39 +0100
Subject: [PATCH 0945/1053] Codefix: possible null pointer dereference

---
 src/misc_gui.cpp | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/misc_gui.cpp b/src/misc_gui.cpp
index d1cc6c404a..2b3643ee78 100644
--- a/src/misc_gui.cpp
+++ b/src/misc_gui.cpp
@@ -1010,13 +1010,15 @@ struct QueryStringWindow : public Window
 
 	void Close([[maybe_unused]] int data = 0) override
 	{
-		if (this->parent->window_class == WC_STATION_VIEW) SetViewportStationRect(Station::Get(this->parent->window_number), false);
-		if (this->parent->window_class == WC_WAYPOINT_VIEW) SetViewportWaypointRect(Waypoint::Get(this->parent->window_number), false);
+		if (this->parent != nullptr) {
+			if (this->parent->window_class == WC_STATION_VIEW) SetViewportStationRect(Station::Get(this->parent->window_number), false);
+			if (this->parent->window_class == WC_WAYPOINT_VIEW) SetViewportWaypointRect(Waypoint::Get(this->parent->window_number), false);
 
-		if (!this->editbox.handled && this->parent != nullptr) {
-			Window *parent = this->parent;
-			this->parent = nullptr; // so parent doesn't try to close us again
-			parent->OnQueryTextFinished(std::nullopt);
+			if (!this->editbox.handled) {
+				Window *parent = this->parent;
+				this->parent = nullptr; // so parent doesn't try to close us again
+				parent->OnQueryTextFinished(std::nullopt);
+			}
 		}
 
 		this->Window::Close();
-- 
2.55.0

