From 27fbcb3cc52afea09da497713aa5eb88aeb5901e Mon Sep 17 00:00:00 2001
From: mmtunligit <156685720+mmtunligit@users.noreply.github.com>
Date: Wed, 17 Dec 2025 09:44:04 +0100
Subject: [PATCH 0951/1053] Fix a41738e: Picker item recolour depends on
 gamemode (#14929)

---
 src/picker_gui.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/picker_gui.cpp b/src/picker_gui.cpp
index 4d02530663..9e5ca47318 100644
--- a/src/picker_gui.cpp
+++ b/src/picker_gui.cpp
@@ -339,8 +339,8 @@ void PickerWindow::DrawWidget(const Rect &r, WidgetID widget) const
 				int by = ir.Height() - ScaleGUITrad(12);
 
 				GrfSpecFeature feature = this->callbacks.GetFeature();
-				/* Houses have recolours but not related to the company colour. */
-				PaletteID palette = feature == GSF_HOUSES ? PAL_NONE : GetCompanyPalette(_local_company);
+				/* Houses have recolours but not related to the company colour and other items depend on gamemode. */
+				PaletteID palette = _game_mode != GM_NORMAL || feature == GSF_HOUSES ? PAL_NONE : GetCompanyPalette(_local_company);
 				DrawBadgeColumn({0, by, ir.Width() - 1, ir.Height() - 1}, 0, this->badge_classes, this->callbacks.GetTypeBadges(item.class_index, item.index), feature, std::nullopt, palette);
 
 				if (this->callbacks.saved.contains(item)) {
@@ -449,8 +449,8 @@ void PickerWindow::OnClick(Point pt, WidgetID widget, int)
 
 		default:
 			if (IsInsideMM(widget, this->badge_filters.first, this->badge_filters.second)) {
-				/* Houses have recolours but not related to the company colour. */
-				PaletteID palette = this->callbacks.GetFeature() == GSF_HOUSES ? PAL_NONE : GetCompanyPalette(_local_company);
+				/* Houses have recolours but not related to the company colour and other items depend on gamemode. */
+				PaletteID palette = _game_mode != GM_NORMAL || this->callbacks.GetFeature() == GSF_HOUSES ? PAL_NONE : GetCompanyPalette(_local_company);
 				ShowDropDownList(this, this->GetWidget<NWidgetBadgeFilter>(widget)->GetDropDownList(palette), -1, widget, 0);
 			}
 			break;
-- 
2.55.0

