From 7e30849c1e019103710288011905f80b5ea862b7 Mon Sep 17 00:00:00 2001
From: Jonathan G Rennison <j.g.rennison@gmail.com>
Date: Fri, 26 Dec 2025 14:06:21 +0000
Subject: [PATCH 0972/1053] Fix b956af63: Incorrect TileIndex used in
 FindNearestGoodCoastalTownSpot (#14988)

---
 src/town_cmd.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/town_cmd.cpp b/src/town_cmd.cpp
index 13d0a8b215..5af5ef9f9a 100644
--- a/src/town_cmd.cpp
+++ b/src/town_cmd.cpp
@@ -2329,7 +2329,7 @@ static TileIndex FindNearestGoodCoastalTownSpot(TileIndex tile, TownLayout layou
 {
 	for (auto coast : SpiralTileSequence(tile, 40)) {
 		/* Find nearest land tile */
-		if (!IsTileType(tile, MP_CLEAR)) continue;
+		if (!IsTileType(coast, MP_CLEAR)) continue;
 
 		TileIndex furthest = INVALID_TILE;
 		uint max_dist = 0;
-- 
2.55.0

