From 20477ff5419c7aa98d2693998215dfc6bb1165f7 Mon Sep 17 00:00:00 2001
From: Michael Lutz <michi@icosahedron.de>
Date: Wed, 31 Dec 2025 23:31:24 +0100
Subject: [PATCH 0980/1053] Codechange: Disable asserts for release (#14883)

---
 cmake/Options.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Options.cmake b/cmake/Options.cmake
index 38d708c717..da2d172808 100644
--- a/cmake/Options.cmake
+++ b/cmake/Options.cmake
@@ -56,7 +56,7 @@ function(set_options)
 
     option(OPTION_DEDICATED "Build dedicated server only (no GUI)" OFF)
     option(OPTION_INSTALL_FHS "Install with Filesystem Hierarchy Standard folders" ${DEFAULT_OPTION_INSTALL_FHS})
-    option(OPTION_USE_ASSERTS "Use assertions; leave enabled for nightlies, betas, and RCs" ON)
+    option(OPTION_USE_ASSERTS "Use assertions; leave enabled for nightlies, betas, and RCs" OFF)
     option(OPTION_USE_NSIS "Use NSIS to create windows installer; enable only for stable releases" OFF)
     option(OPTION_TOOLS_ONLY "Build only tools target" OFF)
     option(OPTION_DOCS_ONLY "Build only docs target" OFF)
-- 
2.55.0

