From bd9e7898a463f74d2f2b2c009e3dd6ae0a1ed661 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Sat, 27 Dec 2025 15:50:32 +0100
Subject: [PATCH 0981/1053] Fix: potential division by 0 when drawing a line
 (#14994)

---
 src/gfx.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gfx.cpp b/src/gfx.cpp
index 950ef25484..9d05ce8d86 100644
--- a/src/gfx.cpp
+++ b/src/gfx.cpp
@@ -341,6 +341,9 @@ static inline void GfxDoDrawLine(void *video, int x, int y, int x2, int y2, int
 		margin  *= 2; // account for rounding errors
 	}
 
+	/* Prevent division by zero. */
+	if (grade_x == 0) grade_x = 1;
+
 	/* Imagine that the line is infinitely long and it intersects with
 	 * infinitely long left and right edges of the clipping rectangle.
 	 * If both intersection points are outside the clipping rectangle
-- 
2.55.0

