From 7f2e4257a39e6bf15da4d9011e19d49d22ea02a6 Mon Sep 17 00:00:00 2001
From: Tyler Trahan <tyler@tylertrahan.com>
Date: Sat, 27 Dec 2025 13:59:28 -0500
Subject: [PATCH 0982/1053] Fix: Use correct 'minutes per year' setting for old
 savegames (#14995)

---
 src/saveload/afterload.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/saveload/afterload.cpp b/src/saveload/afterload.cpp
index 24416e4bba..2f9e10c203 100644
--- a/src/saveload/afterload.cpp
+++ b/src/saveload/afterload.cpp
@@ -735,6 +735,9 @@ bool AfterLoadGame()
 	/* Only new games can use wallclock units. */
 	if (IsSavegameVersionBefore(SLV_ECONOMY_MODE_TIMEKEEPING_UNITS)) _settings_game.economy.timekeeping_units = TKU_CALENDAR;
 
+	/* Set the correct default for 'minutes per year' if the savegame was created before the setting existed. */
+	if (IsSavegameVersionBefore(SLV_CALENDAR_SUB_DATE_FRACT)) _settings_game.economy.minutes_per_calendar_year = CalendarTime::DEF_MINUTES_PER_YEAR;
+
 	/* Update economy year. If we don't have a separate economy date saved, follow the calendar date. */
 	if (IsSavegameVersionBefore(SLV_ECONOMY_DATE)) {
 		TimerGameEconomy::SetDate(TimerGameEconomy::Date{TimerGameCalendar::date.base()}, TimerGameCalendar::date_fract);
-- 
2.55.0

