From 3f5d6c493019c36ce3cd4e2c2a0197beb2724cc6 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Sat, 3 Jan 2026 15:27:39 +0000
Subject: [PATCH 0993/1053] fix: Use correct function to uncheck button

---
 src/settings_gui.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/settings_gui.cpp b/src/settings_gui.cpp
index 33dd58e8cf..fdab8bf2f0 100644
--- a/src/settings_gui.cpp
+++ b/src/settings_gui.cpp
@@ -1653,8 +1653,8 @@ struct GameOptionsWindow : Window {
 		this->SetWidgetDisabledState(WID_GO_GUI_FONT_AA, _fcsettings.prefer_sprite);
 #endif /* HAS_TRUETYPE_FONT */
 
-		this->SetWidgetDisabledState(WID_GO_MOUSE_CURSOR, _settings_client.gui.draw_mouse_cursor);
-		this->SetWidgetDisabledState(WID_GO_WINDOWS_TITLEBARS, _settings_client.gui.windows_titlebars);
+		this->SetWidgetLoweredState(WID_GO_MOUSE_CURSOR, _settings_client.gui.draw_mouse_cursor);
+		this->SetWidgetLoweredState(WID_GO_WINDOWS_TITLEBARS, _settings_client.gui.windows_titlebars);
 
 
 		this->SetWidgetDisabledState(WID_GO_BASE_GRF_DROPDOWN, _game_mode != GM_MENU);
-- 
2.55.0

