From fb981a130fde5469843c47fffc50c51e3c5fc20a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Guilloux?= <glx22@users.noreply.github.com>
Date: Wed, 4 Feb 2026 00:09:02 +0100
Subject: [PATCH 1040/1053] Fix #15225: [CMake] Detect mismatch between tag and
 internal version (#15226)

---
 cmake/scripts/FindVersion.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/scripts/FindVersion.cmake b/cmake/scripts/FindVersion.cmake
index 995d1d124d..cbfb466b52 100644
--- a/cmake/scripts/FindVersion.cmake
+++ b/cmake/scripts/FindVersion.cmake
@@ -135,6 +135,9 @@ if(GENERATE_OTTDREV)
     message(STATUS "Generating .ottdrev")
     file(WRITE ${CMAKE_SOURCE_DIR}/.ottdrev "${REV_VERSION}\t${REV_ISODATE}\t${REV_MODIFIED}\t${REV_HASH}\t${REV_ISTAG}\t${REV_ISSTABLETAG}\n")
 else()
+    if(REV_ISSTABLETAG AND NOT (REV_VERSION STREQUAL "${REV_MAJOR}.${REV_MINOR}"))
+        message(FATAL_ERROR "Tag (${REV_VERSION}) doesn't match internal version (${REV_MAJOR}.${REV_MINOR})")
+    endif()
     message(STATUS "Generating rev.cpp")
     configure_file("${CMAKE_SOURCE_DIR}/src/rev.cpp.in"
             "${FIND_VERSION_BINARY_DIR}/rev.cpp")
-- 
2.55.0

