From c91185ebd95f4bd8356465ee8a732aa058beb745 Mon Sep 17 00:00:00 2001
From: Peter Nelson <peter1138@openttd.org>
Date: Wed, 18 Feb 2026 11:35:32 +0000
Subject: [PATCH 1049/1053] Doc: bump release version to 15.2

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 60f4bc43ff..88fc01a741 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ if(NOT BINARY_NAME)
 endif()
 
 project(${BINARY_NAME}
-    VERSION 15.0
+    VERSION 15.2
     LANGUAGES CXX
 )
 
-- 
2.55.0

