From 8e581bb985462f2efeff0ab71883d624964164c4 Mon Sep 17 00:00:00 2001
From: Miguel Horta <geral@miguelhorta.name>
Date: Sat, 3 Jan 2026 17:42:34 +0000
Subject: [PATCH 1053/1053] Codefix: Use active_tab instead of hardcoded value

---
 src/settings_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/settings_gui.cpp b/src/settings_gui.cpp
index defc319353..46f4c267c4 100644
--- a/src/settings_gui.cpp
+++ b/src/settings_gui.cpp
@@ -926,7 +926,7 @@ struct GameOptionsWindow : Window {
 
 	void OnPaint() override
 	{
-		if (this->GetWidget<NWidgetStacked>(WID_GO_TAB_SELECTION)->shown_plane != 4) {
+		if (GameOptionsWindow::active_tab != WID_GO_TAB_ADVANCED) {
 			this->DrawWidgets();
 			return;
 		}
-- 
2.55.0

